---
author: "Laurine Seelt"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
biblio-style: apalike
link-citations: yes
---

# Free space

This was the fifth assignment for our portfolio. We had/have to fill 32 hours learning something new that will be useful for our internship next year and possibly our job.



- Uitzoeken welke automatiseringen er op microbiologische labs worden gebruikt.
- Metagenomics? 
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03933-4 (megaR package)
http://web.evolbio.mpg.de/~wang/Site/R_tutorial_files/16S%20Metagenomic%20Analysis%20Tutorial.pdf (random tutorial)
https://www.mcs.anl.gov/~braithwaite/matR/docs-and-pubs/matR-user-manual.pdf (matR package)
https://rpubs.com/jrandall7/EICC16s (phyloseq)
https://cran.r-project.org/web/packages/microbial/microbial.pdf (microbial package)
https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html (aritkel over microbioom analyse workflow)

https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/ (soort course over de analyse van microbioom data)
https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html (ook een course over de analyse van microbiota data)
https://grunwaldlab.github.io/analysis_of_microbiome_community_data_in_r/04--manipulating.html (nog een course)
- Python?

## Course: Introduction to the statistical analysis of microbiome data in R
Link to the course: https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

In this course, we examined the differences in microbiota between patients with and without chronic fatigue syndrome.
The aspects that were looked at were: <br>
Taxonomic relative abundance <br>
Hierarchal clustering <br>
Alpha-diversity (microbiome diversity in just one sample) <br>
Beta-divesity (microbiome diversity in two or more samples) <br>
Diffrential abundance testing <br>
Predicting class labels <br>

```{r installing all the required packages, include=TRUE, message=FALSE, warning=FALSE, eval=FALSE}
.cran_packages <- c("tidyverse", "cowplot", "picante", "vegan", "HMP", "dendextend", "rms", "devtools")
.bioc_packages <- c("phyloseq", "DESeq2", "microbiome", "metagenomeSeq", "ALDEx2")
.inst <- .cran_packages %in% installed.packages()
if(any(!.inst)) {
  install.packages(.cran_packages[!.inst])
}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(.bioc_packages, version = "3.15") #Changed 3.9 to 3.15
devtools::install_github("adw96/breakaway")
devtools::install_github(repo = "malucalle/selbal") #changed UVic-omics to malucalle
```

```{r loading the required packages, message=FALSE, warning=FALSE}
library(tidyverse)
library(phyloseq)
library(DESeq2)
library(microbiome)
library(vegan)
library(picante)
library(ALDEx2)
library(metagenomeSeq)
library(HMP)
library(dendextend)
library(selbal)
library(rms)
library(breakaway)
```

```{r, message=FALSE, warning=FALSE}
# Reading in the data
ps <- readRDS("Data/Data_raw/ps_giloteaux_2016.rds")
# Sorting the samples on total read count
sort(phyloseq::sample_sums(ps))
# Removing (samples with?) less than 5000 reads
(ps <- phyloseq::subset_samples(ps, phyloseq::sample_sums(ps) > 5000))
# Removing the OTU's that were only present in the removed samples
(ps <- phyloseq::prune_taxa(phyloseq::taxa_sums(ps) > 0, ps))
# Assigning a new metadata field
phyloseq::sample_data(ps)$Status <- ifelse(phyloseq::sample_data(ps)$Subject == "Patient", "Chronic Fatique", "Control")
phyloseq::sample_data(ps)$Status <- factor(phyloseq::sample_data(ps)$Status, levels = c("Control", "Chronic Fatique"))
ps %>% sample_data %>% dplyr::count(Status)
```

```{r, message=FALSE, warning=FALSE}
# Counting all the phyla in the dataset
table(phyloseq::tax_table(ps)[, "Phylum"])
# Convert the samples to relative abundances
ps_rel_abund = phyloseq::transform_sample_counts(ps, function(x){x / sum(x)})
phyloseq::otu_table(ps)[1:5, 1:5] # Before we converted it to relative abundances
phyloseq::otu_table(ps_rel_abund)[1:5, 1:5] # After we calculated the relative abundances

# Plotting the relative abundances, devided by the two groups
phyloseq::plot_bar(ps_rel_abund, fill = "Phylum")+
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")+
  labs(x = "", y = "Relative Abundance\n")+
  facet_wrap(~ Status, scales = "free")+
  theme(panel.background = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

```{r extra, message=FALSE, warning=FALSE, eval=FALSE}
# Sorting the taxa by their relative abundance
sort(ps_rel_abund@tax_table[["Phylum"]], decreasing = TRUE) %>%  phyloseq::plot_bar(fill = "Phylum")+
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")+
  labs(x = "", y = "Relative Abundance\n")+
  facet_wrap(~ Status, scales = "free")+
  theme(panel.background = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

```{r generating boxplots according to the groups, message=FALSE, warning=FALSE}
# Grouping the taxa according to phylum and renaming the groups
ps_phylum <- phyloseq::tax_glom(ps, "Phylum")
phyloseq::taxa_names(ps_phylum) <- phyloseq::tax_table(ps_phylum)[, "Phylum"]
phyloseq::otu_table(ps_phylum)[1:5, 1:5]

# Melting the data and making the boxplots
phyloseq::psmelt(ps_phylum) %>%
  ggplot(data = ., aes(x = Status, y = Abundance))+
  geom_boxplot(outlier.shape = NA)+
  geom_jitter(aes(colour = OTU), height = 0, width = 0.2)+
  labs(x = "", y = "Abundance\n")+
  facet_wrap(~ OTU, scales = "free")
```

```{r looking at the frequencies of the taxa in both groups, message=FALSE, warning=FALSE}
# Making two subsets
controls <- phyloseq::subset_samples(ps_phylum, Status == "Control")
cf <- phyloseq::subset_samples(ps_phylum, Status = "Chronic Fatigue")

# Looking at the OTU tables
control_otu <- data.frame(phyloseq::otu_table(controls))
cf_otu <- data.frame(phyloseq::otu_table(cf))

# Grouping the rare phyla to improve testing
control_otu <- control_otu %>% t(.) %>% as.data.frame(.) %>% mutate(Other = Cyanobacteria + Euryarchaeota + Tenericutes + Verrucomicrobia + Fusobacteria) %>% dplyr::select(-Cyanobacteria, -Euryarchaeota, -Tenericutes, -Verrucomicrobia, -Fusobacteria)
cf_otu <- cf_otu %>% t(.) %>% as.data.frame(.) %>% mutate(Other = Cyanobacteria + Euryarchaeota + Tenericutes + Verrucomicrobia + Fusobacteria) %>% dplyr::select(-Cyanobacteria, -Euryarchaeota, -Tenericutes, -Verrucomicrobia, -Fusobacteria)

# Performing the HMP test
group_data <- list(control_otu, cf_otu)
(xdc <- HMP::Xdc.sevsample(group_data))

1 - pchisq(0.2769004, 5) # Looking at the Chi-Squared distribution
```

Since the P-value of the HMP test is nearly 1, we cannot reject the null hypothesis that there is no statistical significant distribution of phyla between the two groups.

__Looking at the hierarchical clustering in the samples__
<br>
```{r looking at hierarchical clustering, message=FALSE, warning=FALSE}
# Extracting the OTU table
ps_rel_otu <- data.frame(phyloseq::otu_table(ps_rel_abund))
ps_rel_otu <- t(ps_rel_otu)
# Computing the Bray-Curtis dissimilarity
bc_dist <- vegan::vegdist(ps_rel_otu, method = "bray")
# Generating a matrix with the dissimilarities
as.matrix(bc_dist)[1:5, 1:5]
# Saving it as a dendrogram
ward <- as.dendrogram(hclust(bc_dist, method = "ward.D2"))
# Adding color coding
meta <- data.frame(phyloseq::sample_data(ps_rel_abund))
colorCode <- c(Control = "red", `Chronic Fatigue` = "blue")
labels_colors(ward) <- colorCode[meta$Status][order.dendrogram(ward)]
# Plotting the dendrogram
plot(ward)
```

```{r extra trying to create heatmaps for the two different groups, message=FALSE, warning=FALSE, eval=FALSE}
plot_heatmap(cf)
plot_heatmap(controls)
```

__Looking at the alpha diversity__
<br>
Alpha diversity looks at the diversity _within_ a sample and how the observed OTU's are distributed.
```{r, message=FALSE, warning=FALSE}
# Plotting the observed richness
ggplot(data = data.frame("total_reads" = phyloseq::sample_sums(ps),
                         "observed" = phyloseq::estimate_richness(ps, measures = "Observed")[, 1]),
       aes(x = total_reads, y = observed)) +
  geom_point() +
  geom_smooth(method="lm", se = FALSE) +
  labs(x = "\nTotal Reads", y = "Observed richness\n")
```

Now we're going to subsample the reads, plot them and test for group differences
<br>
```{r, message=FALSE, warning=FALSE}
# Subsample reads
(ps_rare <- phyloseq::rarefy_even_depth(ps, rngseed = 123, replace = FALSE))
head(phyloseq::sample_sums(ps_rare))
# Making a dataframe with the adiv measures, another package to analyse and measure biodiversity
adiv <- data.frame("Observed" = phyloseq::estimate_richness(ps_rare, measures = "Observed"),
"Shannon" = phyloseq::estimate_richness(ps_rare, measures = "Shannon"),
"PD" = picante::pd(samp = data.frame(t(data.frame(phyloseq::otu_table(ps_rare)))), include.root = FALSE, tree=phyloseq::phy_tree(ps_rare))[, 1],
"Status" = phyloseq::sample_data(ps_rare)$Status) # Zelf "include.root = FALSE" toegevoegd, anders kreeg ik steeds een error
head(adiv)
# Plotting the adiv measures
adiv %>% gather(key = metric, value = value, c("Observed", "Shannon", "PD")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "PD"))) %>%
  ggplot(aes(x = Status, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Status), height = 0, width = 0.2) +
  labs(x = "", y = "") +
  facet_wrap(~metric, scales = "free") +
  theme(legend.position = "none")
# Summarising the data
adiv %>% group_by(Status) %>% 
  dplyr::summarise(median_observed = median(Observed),
                   median_shannon = median(Shannon),
                   median_pd = median(PD))
# Performing the Wilcoxon tests for the three estimates
wilcox.test(Observed ~ Status, data = adiv, exact = FALSE, conf.int = TRUE)
wilcox.test(Shannon ~ Status, data = adiv, conf.int = TRUE)
wilcox.test(PD ~ Status, data = adiv, conf.int = TRUE)
```

__Looking at the beta diversity__
<br>
Beta diversity looks at the diversity _between_ samples and how it's similar or dissimilar.
```{r,message=FALSE, warning=FALSE}
# Transforming the Total Read Counts according to the Centered Log Ratio method
(ps_clr <- microbiome::transform(ps, "clr"))
phyloseq::otu_table(ps)[1:5, 1:5] # Old OTU table
phyloseq::otu_table(ps_clr)[1:5, 1:5] # New OTU table
# The value are now the dominance for each taxa relative to the geometric mean of all taxa on the logarithmic scale
```
<br>
<br>

## Course: Microbiota analysis in R
Link to the course: https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html
<br>
<br>
In this course, we will be looking at the fecal bacterial microbiota of 8 calves at ages 2 weeks, 8 weeks and 1 year old and correlating them with variables such as weight gain, expressed as Average Daily Gain in Kilograms(ADGKG), and gastrointestinal short chain fatty acids, SCFA.
<br>
```{r downloading and loading the packages, message=FALSE, warning=FALSE}
# install.packages("ape")
library(ape)
library(dplyr)
library(ggplot2)
library(gplots)
library(lme4)
library(phangorn)
library(plotly)
library(tidyr)
library(here)
library(vegan)
# install.packages("VennDiagram")
library(VennDiagram)
# Installed Java from: https://www.java.com/en/download/manual.jsp
# Installed Java JDk from: https://www.oracle.com/java/technologies/downloads/#jdk18-windows
# install.packages("rJava")
library(rJava)
# install.packages("venneuler")
library(venneuler)
# if (!require("BiocManager", quietly = TRUE))
    # install.packages("BiocManager")
# BiocManager::install(version = "3.15") # Instead of source("https://bioconductor.org/biocLite.R")
# BiocManager::install("phyloseq") # Instead of biocLite("phyloseq") ?
```

```{r reading in the data, message=FALSE, warning=FALSE}
# Downloaded the data from: https://github.com/kdillmcfarland/workshops_UW_Madison/tree/master/Microbiota_analysis_R/Data
OTU = read.table("Data/example.final.an.unique_list.0.03.norm.shared.txt", header = TRUE, sep = "\t")
tax = read.table("Data/example.final.an.unique_list.0.03.cons.taxonomy.txt", header = TRUE, sep = "\t")
meta = read.table("Data/example.metadata.txt", header = TRUE, row.names = 1, sep = "\t")
SCFA = read.table("Data/example.SCFA.txt", header = TRUE, row.names = 1, sep = "\t")
```

```{r data wrangling, message=FALSE, warning=FALSE}
# Setting the "Group" column as row names in the OTU dataset
row.names(OTU) = OTU$Group
# Removing the columns that are not OTU counts
OTU.clean = OTU[,-which(names(OTU) %in% c("label", "numOtus", "Group"))]
# Setting the "OTU" column as row names in the taxonomy table
row.names(tax) = tax$OTU
# Removing the OTU's that aren't present in the OTU.clean dataset
tax.clean = tax[row.names(tax) %in% colnames(OTU.clean),]
# Separating the taxonomy table so each level has its own column
tax.clean = separate(tax.clean, Taxonomy, into = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species", "Strain"), sep = ";")
# Removing the "Size", "Strain" and "OTU" columns because these are now row names
tax.clean = tax.clean[,-which(names(tax.clean) %in% c("Size", "Strain", "OTU"))]
# Making sure the three datasets have samples in the same order
OTU.clean = OTU.clean[order(row.names(OTU.clean)),]
meta = meta[order(row.names(meta)),]
SCFA = SCFA[order(row.names(SCFA)),]
# set.seed to make the analysis reproducible
set.seed(8765)
```

__Looking at the alpha diversity__ <br>
Alpha diversity is the diversity _within_ a sample. It looks at richness, which is the _amount_ of OTU's in each sample, and at evenness, which is how evenly the different OTU's are distributed within the sample. <br>
```{r looking at the alpha diversity, message=FALSE, warning=FALSE}
# Creating a 2x2 plot environment so we can see all 4 metrics at once
par(mfrow = c(2,2))
# Plotting the four metrics
hist(meta$shannon, main = "Shannon diversity", xlab = "", breaks = 10)
hist(meta$simpson, main = "Simpson diversity", xlab = "", breaks = 10)
hist(meta$chao, main = "Chao richness", xlab = "", breaks = 15)
hist(meta$ace, main = "ACE richness", xlab = "", breaks = 15)
```
<br>
None of the data are normally distributed. Simpson diversity is very often skewed as seen in this histogram, so we'll calculate 1/Simpson and plot the metrics again.
<br>
```{r, message=FALSE, warning=FALSE}
# Creating a 2x2 environment
par(mfrow = c(2,2))
# Plotting the four metrics again
hist(meta$shannon, main = "Shannon diversity", xlab = "", breaks = 10)
hist(1/meta$simpson, main = "Inverse Simpson diversity", xlab = "", breaks = 10)
hist(meta$chao, main = "Chao richness", xlab = "", breaks = 15)
hist(meta$ace, main = "ACE richness", xlab = "", breaks = 15)
```
<br>
Now the Simpson diversity is distributed similarly to the other richness metrics.
<br>
Next, we'll test the four metrics for normal distribution.
<br>
```{r normality tests, message=FALSE, warning=FALSE}
shapiro.test(meta$shannon)
shapiro.test(1/meta$simpson)
shapiro.test(meta$chao)
shapiro.test(meta$ace)
```
<br>
None of the richness metrics are normally distributed, which was to be expected from the graphs we've seen. So we cannot run any tests that assume the data is normally distributed.
<br>
For illustration purposes, we'll run the ANOVA test with the Shannon's diversity because that's the closest to normally distributed. We'll look at if age impacts the Shannon diversity of the fecal microbiota.
<br>
```{r Anova, message=FALSE, warning=FALSE}
aov.shannon.age = aov(shannon ~ AgeGroup, data = meta)
summary(aov.shannon.age)
```
<br>
We'll run Tukey's honest significance test to do pairwise comparisons between groups and correct for multiple comparisons.
<br>
```{r TukeyHSD, message=FALSE, warning=FALSE}
TukeyHSD(aov.shannon.age)
```
<br>
It's clear that all age groups have significantly different diversity. In a plot, we can clearly see that _diversity_ increases with ages.
<br>
```{r plotting the diversity, message=FALSE, warning=FALSE}
# Re-ordering the groups
meta$AgeGroup.ord = factor(meta$AgeGroup, c("2w", "8w", "1yr"))
# Returning the plot area to 1x1
par(mfrow = c(1,1))
# Plotting the diversity
boxplot(shannon ~ AgeGroup.ord, data = meta, 
        ylab = "Shannon's diversity",
        xlab = "Age group") # Added the xlab myself because it looks better :)
```
<br>
To illustrate some non-parametric tests, we'll use Chao's richness estimate. Age is categorical, so we'll use Kruskal-Wallis.
<br>
```{r Kruskal-Wallis, message=FALSE, warning=FALSE}
kruskal.test(chao ~ AgeGroup, data = meta)
```
<br>
We can also test pairwise within age groups with Wilcoxon Rank Sum Tests.
<br>
```{r Wilcoxon Rank Sum Tests, message=FALSE, warning=FALSE}
pairwise.wilcox.test(meta$chao, meta$AgeGroup, p.adjust.method = "fdr")
```
<br>
Just like the diversity, __richness__ also increases with age.
<br>
```{r plotting the richness, message=FALSE, warning=FALSE}
# Creating a 1x1 plot environment
par(mfrow = c(1,1))
# Plotting the richness
boxplot(chao ~ AgeGroup.ord, data = meta,
        ylab = "Chao richness",
        xlab = "Age group") # Again, added the xlab myself for aesthetics :)
```
<br>
Average Daily Gain is a continuous variable, so we'll use a linear model to visualise it.
<br>
We'll use Shannon's diversity again to run some tests that are meant for normally distributed data. We'll take a look at if the ADG impacts the Shannon diversity of the fecal microbiota.
<br>
```{r glm and lm, message=FALSE, warning=FALSE}
glm.shannon.ADG = glm(shannon ~ ADGKG, data = meta)
summary(glm.shannon.ADG)
```
<br>
This shows that the intercept of our model is signficantly different from 0, bu the slope is not, and the slope is our variable of interest.
<br>
Next, let's plot the glm.
<br>
```{r plotting the glm, message=FALSE, warning=FALSE}
plot(shannon ~ ADGKG, data = meta)
abline(glm.shannon.ADG)
```
<br>
To illustrate non-normally distributed data, we'll use Chao's richness estimate again. We'll first use the Gaussian distribution, but we already know this isn't a good fit.
<br>
```{r plotting Gaussian, message=FALSE, warning=FALSE}
gaussian.chao.adg = glm(chao ~ADGKG, data = meta, family = "gaussian")
par(mfrow = c(1,2))
plot(gaussian.chao.adg, which = c(1,2))
```
<br>
Next, we'll plot the quasipoisson distribuiton.
<br>
```{r plotting quasipoisson, message=FALSE, warning=FALSE}
qp.chao.ADG = glm(chao ~ ADGKG, data = meta, family = "quasipoisson")
par(mfrow = c(1,2))
plot(qp.chao.ADG, which = c(1,2))
```
<br>
The quasipoisson distribution fits much better than the Gaussian distribution, so we'll use quasipoisson for further calculations.
<br>
```{r, message=FALSE, warning=FALSE}
summary(qp.chao.ADG)
```
<br>
We'll plot Chao and ADGKG again, to look at the correlation.
<br>
```{r plotting Chao and ADGKG, message=FALSE, warning=FALSE}
par(mfrow = c(1,1))
plot(log(chao) ~ ADGKG, data = meta, ylab = "ln(Chao's richness)")
abline(qp.chao.ADG)
```
<br>
Looking at the graph, there is nog significant correlation between Chao's richness and the Average Daily Gain.
