---
author: "Laurine Seelt"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
biblio-style: apalike
link-citations: yes
---

# Free space

This was the fifth assignment for our portfolio. We had/have to fill 32 hours learning something new that will be useful for our internship next year and possibly our job.



- Uitzoeken welke automatiseringen er op microbiologische labs worden gebruikt.
- Metagenomics? 
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03933-4 (megaR package)
http://web.evolbio.mpg.de/~wang/Site/R_tutorial_files/16S%20Metagenomic%20Analysis%20Tutorial.pdf (random tutorial)
https://www.mcs.anl.gov/~braithwaite/matR/docs-and-pubs/matR-user-manual.pdf (matR package)
https://rpubs.com/jrandall7/EICC16s (phyloseq)
https://cran.r-project.org/web/packages/microbial/microbial.pdf (microbial package)
https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html (aritkel over microbioom analyse workflow)

https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/ (soort course over de analyse van microbioom data)
https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html (ook een course over de analyse van microbiota data)
https://grunwaldlab.github.io/analysis_of_microbiome_community_data_in_r/04--manipulating.html (nog een course)
- Python?

## Course: Introduction to the statistical analysis of microbiome data in R
In this course, we examined the differences in microbiota between patients with and without chronic fatigue syndrome.
The aspects that were looked at were: <br>
Taxonomic relative abundance <br>
Hierarchal clustering <br>
Alpha-diversity (microbiome diversity in just one sample) <br>
Beta-divesity (microbiome diversity in two or more samples) <br>
Diffrential abundance testing <br>
Predicting class labels <br>

```{r installing all the required packages, include=TRUE, message=FALSE, warning=FALSE, eval=FALSE}
.cran_packages <- c("tidyverse", "cowplot", "picante", "vegan", "HMP", "dendextend", "rms", "devtools")
.bioc_packages <- c("phyloseq", "DESeq2", "microbiome", "metagenomeSeq", "ALDEx2")
.inst <- .cran_packages %in% installed.packages()
if(any(!.inst)) {
  install.packages(.cran_packages[!.inst])
}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(.bioc_packages, version = "3.14") #Changed 3.9 to 3.14
devtools::install_github("adw96/breakaway")
devtools::install_github(repo = "malucalle/selbal") #changed UVic-omics to malucalle
```

```{r loading the required packages, message=FALSE, warning=FALSE}
library(tidyverse)
library(phyloseq)
library(DESeq2)
library(microbiome)
library(vegan)
library(picante)
library(ALDEx2)
library(metagenomeSeq)
library(HMP)
library(dendextend)
library(selbal)
library(rms)
library(breakaway)
```

```{r, message=FALSE, warning=FALSE}
# Reading in the data
ps <- readRDS("Data/Data_raw/ps_giloteaux_2016.rds")
# Sorting the samples on total read count
sort(phyloseq::sample_sums(ps))
# Removing (samples with?) less than 5000 reads
(ps <- phyloseq::subset_samples(ps, phyloseq::sample_sums(ps) > 5000))
# Removing the OTU's that were only present in the removed samples
(ps <- phyloseq::prune_taxa(phyloseq::taxa_sums(ps) > 0, ps))
# Assigning a new metadata field
phyloseq::sample_data(ps)$Status <- ifelse(phyloseq::sample_data(ps)$Subject == "Patient", "Chronic Fatique", "Control")
phyloseq::sample_data(ps)$Status <- factor(phyloseq::sample_data(ps)$Status, levels = c("Control", "Chronic Fatique"))
ps %>% sample_data %>% dplyr::count(Status)
```

```{r, message=FALSE, warning=FALSE}
# Counting all the phyla in the dataset
table(phyloseq::tax_table(ps)[, "Phylum"])
# Convert the samples to relative abundances
ps_rel_abund = phyloseq::transform_sample_counts(ps, function(x){x / sum(x)})
phyloseq::otu_table(ps)[1:5, 1:5] # Before we converted it to relative abundances
phyloseq::otu_table(ps_rel_abund)[1:5, 1:5] # After we calculated the relative abundances

# Plotting the relative abundances, devided by the two groups
phyloseq::plot_bar(ps_rel_abund, fill = "Phylum")+
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")+
  labs(x = "", y = "Relative Abundance\n")+
  facet_wrap(~ Status, scales = "free")+
  theme(panel.background = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

```{r extra, message=FALSE, warning=FALSE, eval=FALSE}
# Sorting the taxa by their relative abundance
sort(ps_rel_abund@tax_table[["Phylum"]], decreasing = TRUE) %>%  phyloseq::plot_bar(fill = "Phylum")+
  geom_bar(aes(color = Phylum, fill = Phylum), stat = "identity", position = "stack")+
  labs(x = "", y = "Relative Abundance\n")+
  facet_wrap(~ Status, scales = "free")+
  theme(panel.background = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
```

```{r generating boxplots according to the groups, message=FALSE, warning=FALSE}
# Grouping the taxa according to phylum and renaming the groups
ps_phylum <- phyloseq::tax_glom(ps, "Phylum")
phyloseq::taxa_names(ps_phylum) <- phyloseq::tax_table(ps_phylum)[, "Phylum"]
phyloseq::otu_table(ps_phylum)[1:5, 1:5]

# Melting the data and making the boxplots
phyloseq::psmelt(ps_phylum) %>%
  ggplot(data = ., aes(x = Status, y = Abundance))+
  geom_boxplot(outlier.shape = NA)+
  geom_jitter(aes(colour = OTU), height = 0, width = 0.2)+
  labs(x = "", y = "Abundance\n")+
  facet_wrap(~ OTU, scales = "free")
```

```{r looking at the frequencies of the taxa in both groups, message=FALSE, warning=FALSE}
# Making two subsets
controls <- phyloseq::subset_samples(ps_phylum, Status == "Control")
cf <- phyloseq::subset_samples(ps_phylum, Status = "Chronic Fatigue")

# Looking at the OTU tables
control_otu <- data.frame(phyloseq::otu_table(controls))
cf_otu <- data.frame(phyloseq::otu_table(cf))

# Grouping the rare phyla to improve testing
control_otu <- control_otu %>% t(.) %>% as.data.frame(.) %>% mutate(Other = Cyanobacteria + Euryarchaeota + Tenericutes + Verrucomicrobia + Fusobacteria) %>% dplyr::select(-Cyanobacteria, -Euryarchaeota, -Tenericutes, -Verrucomicrobia, -Fusobacteria)
cf_otu <- cf_otu %>% t(.) %>% as.data.frame(.) %>% mutate(Other = Cyanobacteria + Euryarchaeota + Tenericutes + Verrucomicrobia + Fusobacteria) %>% dplyr::select(-Cyanobacteria, -Euryarchaeota, -Tenericutes, -Verrucomicrobia, -Fusobacteria)

# Performing the HMP test
group_data <- list(control_otu, cf_otu)
(xdc <- HMP::Xdc.sevsample(group_data))

1 - pchisq(0.2769004, 5) # Looking at the Chi-Squared distribution
```

Since the P-value of the HMP test is nearly 1, we cannot reject the null hypothesis that there is no statistical significant distribution of phyla between the two groups.

```{r looking at hierarchical clustering, message=FALSE, warning=FALSE}
# Extracting the OTU table
ps_rel_otu <- data.frame(phyloseq::otu_table(ps_rel_abund))
ps_rel_otu <- t(ps_rel_otu)
# Computing the Bray-Curtis dissimilarity
bc_dist <- vegan::vegdist(ps_rel_otu, method = "bray")
# Generating a matrix with the dissimilarities
as.matrix(bc_dist)[1:5, 1:5]
# Saving it as a dendrogram
ward <- as.dendrogram(hclust(bc_dist, method = "ward.D2"))
# Adding color coding
meta <- data.frame(phyloseq::sample_data(ps_rel_abund))
colorCode <- c(Control = "red", `Chronic Fatigue` = "blue")
labels_colors(ward) <- colorCode[meta$Status][order.dendrogram(ward)]
# Plotting the dendrogram
plot(ward)
```

```{r extra trying to create heatmaps for the two different groups, message=FALSE, warning=FALSE, eval=FALSE}
plot_heatmap(cf)
plot_heatmap(controls)
```

__Looking at the alpha diversity__
<br>
Alpha diversity looks at the diversity _within_ a sample and how the observed OTU's are distributed.
```{r}

```

