<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Free space | Portfolio Laurine Seelt</title>
  <meta name="description" content="This is the bookdown for my DSFB2 portfolio" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Free space | Portfolio Laurine Seelt" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the bookdown for my DSFB2 portfolio" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Free space | Portfolio Laurine Seelt" />
  
  <meta name="twitter:description" content="This is the bookdown for my DSFB2 portfolio" />
  

<meta name="author" content="Laurine Seelt" />


<meta name="date" content="2022-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cv.html"/>
<link rel="next" href="using-github-and-zotero.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="investigating-a-c.-elegans-plate-experiment.html"><a href="investigating-a-c.-elegans-plate-experiment.html"><i class="fa fa-check"></i><b>2</b> Investigating a <em>C. Elegans</em> plate experiment</a></li>
<li class="chapter" data-level="3" data-path="reading-two-different-studies-and-recreating-some-of-the-code.html"><a href="reading-two-different-studies-and-recreating-some-of-the-code.html"><i class="fa fa-check"></i><b>3</b> Reading two different studies and recreating some of the code</a>
<ul>
<li class="chapter" data-level="3.1" data-path="reading-two-different-studies-and-recreating-some-of-the-code.html"><a href="reading-two-different-studies-and-recreating-some-of-the-code.html#part-1"><i class="fa fa-check"></i><b>3.1</b> Part 1</a></li>
<li class="chapter" data-level="3.2" data-path="reading-two-different-studies-and-recreating-some-of-the-code.html"><a href="reading-two-different-studies-and-recreating-some-of-the-code.html#part-2"><i class="fa fa-check"></i><b>3.2</b> Part 2</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cleaning-up-according-to-the-guerrilla-analytics-framework.html"><a href="cleaning-up-according-to-the-guerrilla-analytics-framework.html"><i class="fa fa-check"></i><b>4</b> Cleaning up according to the Guerrilla analytics framework</a></li>
<li class="chapter" data-level="5" data-path="cv.html"><a href="cv.html"><i class="fa fa-check"></i><b>5</b> CV</a></li>
<li class="chapter" data-level="6" data-path="free-space.html"><a href="free-space.html"><i class="fa fa-check"></i><b>6</b> Free space</a>
<ul>
<li class="chapter" data-level="6.1" data-path="free-space.html"><a href="free-space.html#course-introduction-to-the-statistical-analysis-of-microbiome-data-in-r"><i class="fa fa-check"></i><b>6.1</b> Course: Introduction to the statistical analysis of microbiome data in R</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="using-github-and-zotero.html"><a href="using-github-and-zotero.html"><i class="fa fa-check"></i><b>7</b> Using GitHub and Zotero</a></li>
<li class="chapter" data-level="8" data-path="making-a-website-on-github.html"><a href="making-a-website-on-github.html"><i class="fa fa-check"></i><b>8</b> Making a website on GitHub</a></li>
<li class="chapter" data-level="9" data-path="working-with-dbeaver-and-sql.html"><a href="working-with-dbeaver-and-sql.html"><i class="fa fa-check"></i><b>9</b> Working with DBeaver and SQL</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio Laurine Seelt</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="free-space" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Free space<a href="free-space.html#free-space" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This was the fifth assignment for our portfolio. We had/have to fill 32 hours learning something new that will be useful for our internship next year and possibly our job.</p>
<ul>
<li>Uitzoeken welke automatiseringen er op microbiologische labs worden gebruikt.</li>
<li>Metagenomics?
<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03933-4" class="uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03933-4</a> (megaR package)
<a href="http://web.evolbio.mpg.de/~wang/Site/R_tutorial_files/16S%20Metagenomic%20Analysis%20Tutorial.pdf" class="uri">http://web.evolbio.mpg.de/~wang/Site/R_tutorial_files/16S%20Metagenomic%20Analysis%20Tutorial.pdf</a> (random tutorial)
<a href="https://www.mcs.anl.gov/~braithwaite/matR/docs-and-pubs/matR-user-manual.pdf" class="uri">https://www.mcs.anl.gov/~braithwaite/matR/docs-and-pubs/matR-user-manual.pdf</a> (matR package)
<a href="https://rpubs.com/jrandall7/EICC16s" class="uri">https://rpubs.com/jrandall7/EICC16s</a> (phyloseq)
<a href="https://cran.r-project.org/web/packages/microbial/microbial.pdf" class="uri">https://cran.r-project.org/web/packages/microbial/microbial.pdf</a> (microbial package)
<a href="https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html" class="uri">https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html</a> (aritkel over microbioom analyse workflow)</li>
</ul>
<p><a href="https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/" class="uri">https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/</a> (soort course over de analyse van microbioom data)
<a href="https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html" class="uri">https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html</a> (ook een course over de analyse van microbiota data)
<a href="https://grunwaldlab.github.io/analysis_of_microbiome_community_data_in_r/04--manipulating.html" class="uri">https://grunwaldlab.github.io/analysis_of_microbiome_community_data_in_r/04--manipulating.html</a> (nog een course)
- Python?</p>
<div id="course-introduction-to-the-statistical-analysis-of-microbiome-data-in-r" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Course: Introduction to the statistical analysis of microbiome data in R<a href="free-space.html#course-introduction-to-the-statistical-analysis-of-microbiome-data-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this course, we examined the differences in microbiota between patients with and without chronic fatigue syndrome.
The aspects that were looked at were: <br>
Taxonomic relative abundance <br>
Hierarchal clustering <br>
Alpha-diversity (microbiome diversity in just one sample) <br>
Beta-divesity (microbiome diversity in two or more samples) <br>
Diffrential abundance testing <br>
Predicting class labels <br></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="free-space.html#cb14-1" aria-hidden="true" tabindex="-1"></a>.cran_packages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;cowplot&quot;</span>, <span class="st">&quot;picante&quot;</span>, <span class="st">&quot;vegan&quot;</span>, <span class="st">&quot;HMP&quot;</span>, <span class="st">&quot;dendextend&quot;</span>, <span class="st">&quot;rms&quot;</span>, <span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb14-2"><a href="free-space.html#cb14-2" aria-hidden="true" tabindex="-1"></a>.bioc_packages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;phyloseq&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>, <span class="st">&quot;microbiome&quot;</span>, <span class="st">&quot;metagenomeSeq&quot;</span>, <span class="st">&quot;ALDEx2&quot;</span>)</span>
<span id="cb14-3"><a href="free-space.html#cb14-3" aria-hidden="true" tabindex="-1"></a>.inst <span class="ot">&lt;-</span> .cran_packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()</span>
<span id="cb14-4"><a href="free-space.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="sc">!</span>.inst)) {</span>
<span id="cb14-5"><a href="free-space.html#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(.cran_packages[<span class="sc">!</span>.inst])</span>
<span id="cb14-6"><a href="free-space.html#cb14-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-7"><a href="free-space.html#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb14-8"><a href="free-space.html#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb14-9"><a href="free-space.html#cb14-9" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(.bioc_packages, <span class="at">version =</span> <span class="st">&quot;3.14&quot;</span>) <span class="co">#Changed 3.9 to 3.14</span></span>
<span id="cb14-10"><a href="free-space.html#cb14-10" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;adw96/breakaway&quot;</span>)</span>
<span id="cb14-11"><a href="free-space.html#cb14-11" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="at">repo =</span> <span class="st">&quot;malucalle/selbal&quot;</span>) <span class="co">#changed UVic-omics to malucalle</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="free-space.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-2"><a href="free-space.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb15-3"><a href="free-space.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb15-4"><a href="free-space.html#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb15-5"><a href="free-space.html#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb15-6"><a href="free-space.html#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(picante)</span>
<span id="cb15-7"><a href="free-space.html#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ALDEx2)</span>
<span id="cb15-8"><a href="free-space.html#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(metagenomeSeq)</span>
<span id="cb15-9"><a href="free-space.html#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(HMP)</span>
<span id="cb15-10"><a href="free-space.html#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dendextend)</span>
<span id="cb15-11"><a href="free-space.html#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(selbal)</span>
<span id="cb15-12"><a href="free-space.html#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rms)</span>
<span id="cb15-13"><a href="free-space.html#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(breakaway)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="free-space.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Reading in the data</span></span>
<span id="cb16-2"><a href="free-space.html#cb16-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;Data/Data_raw/ps_giloteaux_2016.rds&quot;</span>)</span>
<span id="cb16-3"><a href="free-space.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Sorting the samples on total read count</span></span>
<span id="cb16-4"><a href="free-space.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(phyloseq<span class="sc">::</span><span class="fu">sample_sums</span>(ps))</span></code></pre></div>
<pre><code>## ERR1331827 ERR1331852 ERR1331856 ERR1331869 ERR1331833 ERR1331797 ERR1331786 
##       2707       3031       3117       5083       5245       5307       5696 
## ERR1331818 ERR1331792 ERR1331803 ERR1331793 ERR1331819 ERR1331858 ERR1331807 
##       5733       6463       6512       6622       6900       6913       7121 
## ERR1331815 ERR1331821 ERR1331843 ERR1331795 ERR1331846 ERR1331811 ERR1331845 
##       7179       7272       7284       7314       7569       7665       7815 
## ERR1331842 ERR1331838 ERR1331855 ERR1331824 ERR1331832 ERR1331804 ERR1331868 
##       7911       8102       8115       8148       8186       8236       8612 
## ERR1331831 ERR1331859 ERR1331790 ERR1331789 ERR1331837 ERR1331857 ERR1331801 
##       8840       9016       9085       9184       9731       9966      11173 
## ERR1331841 ERR1331861 ERR1331820 ERR1331854 ERR1331863 ERR1331806 ERR1331787 
##      11442      11826      12940      13029      13094      13095      13690 
## ERR1331853 ERR1331851 ERR1331836 ERR1331835 ERR1331802 ERR1331799 ERR1331847 
##      14113      14365      14488      14753      14799      14833      15290 
## ERR1331834 ERR1331817 ERR1331809 ERR1331828 ERR1331813 ERR1331798 ERR1331816 
##      15367      15460      16162      16494      16749      16947      17015 
## ERR1331830 ERR1331785 ERR1331823 ERR1331865 ERR1331848 ERR1331800 ERR1331867 
##      17457      17557      18506      19013      19257      19443      19732 
## ERR1331870 ERR1331810 ERR1331825 ERR1331866 ERR1331871 ERR1331849 ERR1331860 
##      19783      19909      20069      20760      20862      21540      21553 
## ERR1331808 ERR1331872 ERR1331812 ERR1331850 ERR1331791 ERR1331788 ERR1331796 
##      21713      22339      22518      22639      23246      23751      23792 
## ERR1331840 ERR1331826 ERR1331822 ERR1331862 ERR1331864 ERR1331829 ERR1331844 
##      24752      28186      28556      31064      44533      51918      57214 
## ERR1331805 ERR1331839 ERR1331794 
##      59355      61206      65941</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="free-space.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing (samples with?) less than 5000 reads</span></span>
<span id="cb18-2"><a href="free-space.html#cb18-2" aria-hidden="true" tabindex="-1"></a>(ps <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(ps, phyloseq<span class="sc">::</span><span class="fu">sample_sums</span>(ps) <span class="sc">&gt;</span> <span class="dv">5000</span>))</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 138 taxa and 84 samples ]
## sample_data() Sample Data:       [ 84 samples by 22 sample variables ]
## tax_table()   Taxonomy Table:    [ 138 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 138 tips and 136 internal nodes ]
## refseq()      DNAStringSet:      [ 138 reference sequences ]</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="free-space.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing the OTU&#39;s that were only present in the removed samples</span></span>
<span id="cb20-2"><a href="free-space.html#cb20-2" aria-hidden="true" tabindex="-1"></a>(ps <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">prune_taxa</span>(phyloseq<span class="sc">::</span><span class="fu">taxa_sums</span>(ps) <span class="sc">&gt;</span> <span class="dv">0</span>, ps))</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 138 taxa and 84 samples ]
## sample_data() Sample Data:       [ 84 samples by 22 sample variables ]
## tax_table()   Taxonomy Table:    [ 138 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 138 tips and 136 internal nodes ]
## refseq()      DNAStringSet:      [ 138 reference sequences ]</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="free-space.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assigning a new metadata field</span></span>
<span id="cb22-2"><a href="free-space.html#cb22-2" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Subject <span class="sc">==</span> <span class="st">&quot;Patient&quot;</span>, <span class="st">&quot;Chronic Fatique&quot;</span>, <span class="st">&quot;Control&quot;</span>)</span>
<span id="cb22-3"><a href="free-space.html#cb22-3" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Status <span class="ot">&lt;-</span> <span class="fu">factor</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>, <span class="st">&quot;Chronic Fatique&quot;</span>))</span>
<span id="cb22-4"><a href="free-space.html#cb22-4" aria-hidden="true" tabindex="-1"></a>ps <span class="sc">%&gt;%</span> sample_data <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(Status)</span></code></pre></div>
<pre><code>## $Status
## [1] Control         Chronic Fatique
## Levels: Control Chronic Fatique
## 
## $n
## [1] 37 47
## 
## attr(,&quot;row.names&quot;)
## [1] 1 2
## attr(,&quot;class&quot;)
## [1] &quot;sample_data&quot;
## attr(,&quot;class&quot;)attr(,&quot;package&quot;)
## [1] &quot;phyloseq&quot;
## attr(,&quot;.S3Class&quot;)
## [1] &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="free-space.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Counting all the phyla in the dataset</span></span>
<span id="cb24-2"><a href="free-space.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(ps)[, <span class="st">&quot;Phylum&quot;</span>])</span></code></pre></div>
<pre><code>## 
##  Actinobacteria   Bacteroidetes   Cyanobacteria   Euryarchaeota      Firmicutes 
##               7              11               2               1             105 
##    Fusobacteria  Proteobacteria     Tenericutes Verrucomicrobia 
##               1               7               2               1</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="free-space.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the samples to relative abundances</span></span>
<span id="cb26-2"><a href="free-space.html#cb26-2" aria-hidden="true" tabindex="-1"></a>ps_rel_abund <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">transform_sample_counts</span>(ps, <span class="cf">function</span>(x){x <span class="sc">/</span> <span class="fu">sum</span>(x)})</span>
<span id="cb26-3"><a href="free-space.html#cb26-3" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co"># Before we converted it to relative abundances</span></span></code></pre></div>
<pre><code>## OTU Table:          [5 taxa and 5 samples]
##                      taxa are rows
##      ERR1331793 ERR1331872 ERR1331819 ERR1331794 ERR1331851
## OTU1          2        581        347        916      10498
## OTU2        371         46          0        233        301
## OTU3       1189         81        637        199          0
## OTU4          0        172        246          0        372
## OTU5        308         44        143        155        221</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="free-space.html#cb28-1" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_rel_abund)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co"># After we calculated the relative abundances</span></span></code></pre></div>
<pre><code>## OTU Table:          [5 taxa and 5 samples]
##                      taxa are rows
##        ERR1331793  ERR1331872 ERR1331819  ERR1331794 ERR1331851
## OTU1 0.0003020236 0.026008326 0.05028986 0.013891206 0.73080404
## OTU2 0.0560253700 0.002059179 0.00000000 0.003533462 0.02095371
## OTU3 0.1795530051 0.003625946 0.09231884 0.003017849 0.00000000
## OTU4 0.0000000000 0.007699539 0.03565217 0.000000000 0.02589628
## OTU5 0.0465116279 0.001969649 0.02072464 0.002350586 0.01538462</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="free-space.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the relative abundances, devided by the two groups</span></span>
<span id="cb30-2"><a href="free-space.html#cb30-2" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">plot_bar</span>(ps_rel_abund, <span class="at">fill =</span> <span class="st">&quot;Phylum&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-3"><a href="free-space.html#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">color =</span> Phylum, <span class="at">fill =</span> Phylum), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-4"><a href="free-space.html#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Relative Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-5"><a href="free-space.html#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Status, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-6"><a href="free-space.html#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb30-7"><a href="free-space.html#cb30-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb30-8"><a href="free-space.html#cb30-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="free-space.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sorting the taxa by their relative abundance</span></span>
<span id="cb31-2"><a href="free-space.html#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(ps_rel_abund<span class="sc">@</span>tax_table[[<span class="st">&quot;Phylum&quot;</span>]], <span class="at">decreasing =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span>  phyloseq<span class="sc">::</span><span class="fu">plot_bar</span>(<span class="at">fill =</span> <span class="st">&quot;Phylum&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-3"><a href="free-space.html#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">color =</span> Phylum, <span class="at">fill =</span> Phylum), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-4"><a href="free-space.html#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Relative Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-5"><a href="free-space.html#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Status, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-6"><a href="free-space.html#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb31-7"><a href="free-space.html#cb31-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb31-8"><a href="free-space.html#cb31-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="free-space.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Grouping the taxa according to phylum and renaming the groups</span></span>
<span id="cb32-2"><a href="free-space.html#cb32-2" aria-hidden="true" tabindex="-1"></a>ps_phylum <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">tax_glom</span>(ps, <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb32-3"><a href="free-space.html#cb32-3" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">taxa_names</span>(ps_phylum) <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(ps_phylum)[, <span class="st">&quot;Phylum&quot;</span>]</span>
<span id="cb32-4"><a href="free-space.html#cb32-4" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_phylum)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## OTU Table:          [5 taxa and 5 samples]
##                      taxa are rows
##                ERR1331793 ERR1331872 ERR1331819 ERR1331794 ERR1331851
## Bacteroidetes        1903        878       1837       1969      11776
## Proteobacteria        119       3315        468      62358        319
## Firmicutes           4319      14429       3548       1609       2207
## Actinobacteria         30        976         17          0         58
## Cyanobacteria         246          0          0          0          0</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="free-space.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Melting the data and making the boxplots</span></span>
<span id="cb34-2"><a href="free-space.html#cb34-2" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">psmelt</span>(ps_phylum) <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="free-space.html#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data =</span> ., <span class="fu">aes</span>(<span class="at">x =</span> Status, <span class="at">y =</span> Abundance))<span class="sc">+</span></span>
<span id="cb34-4"><a href="free-space.html#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb34-5"><a href="free-space.html#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> OTU), <span class="at">height =</span> <span class="dv">0</span>, <span class="at">width =</span> <span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb34-6"><a href="free-space.html#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>)<span class="sc">+</span></span>
<span id="cb34-7"><a href="free-space.html#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> OTU, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/generating%20boxplots%20according%20to%20the%20groups-1.png" width="672" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="free-space.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making two subsets</span></span>
<span id="cb35-2"><a href="free-space.html#cb35-2" aria-hidden="true" tabindex="-1"></a>controls <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(ps_phylum, Status <span class="sc">==</span> <span class="st">&quot;Control&quot;</span>)</span>
<span id="cb35-3"><a href="free-space.html#cb35-3" aria-hidden="true" tabindex="-1"></a>cf <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(ps_phylum, <span class="at">Status =</span> <span class="st">&quot;Chronic Fatigue&quot;</span>)</span>
<span id="cb35-4"><a href="free-space.html#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="free-space.html#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Looking at the OTU tables</span></span>
<span id="cb35-6"><a href="free-space.html#cb35-6" aria-hidden="true" tabindex="-1"></a>control_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(controls))</span>
<span id="cb35-7"><a href="free-space.html#cb35-7" aria-hidden="true" tabindex="-1"></a>cf_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(cf))</span>
<span id="cb35-8"><a href="free-space.html#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="free-space.html#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Grouping the rare phyla to improve testing</span></span>
<span id="cb35-10"><a href="free-space.html#cb35-10" aria-hidden="true" tabindex="-1"></a>control_otu <span class="ot">&lt;-</span> control_otu <span class="sc">%&gt;%</span> <span class="fu">t</span>(.) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Other =</span> Cyanobacteria <span class="sc">+</span> Euryarchaeota <span class="sc">+</span> Tenericutes <span class="sc">+</span> Verrucomicrobia <span class="sc">+</span> Fusobacteria) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Cyanobacteria, <span class="sc">-</span>Euryarchaeota, <span class="sc">-</span>Tenericutes, <span class="sc">-</span>Verrucomicrobia, <span class="sc">-</span>Fusobacteria)</span>
<span id="cb35-11"><a href="free-space.html#cb35-11" aria-hidden="true" tabindex="-1"></a>cf_otu <span class="ot">&lt;-</span> cf_otu <span class="sc">%&gt;%</span> <span class="fu">t</span>(.) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Other =</span> Cyanobacteria <span class="sc">+</span> Euryarchaeota <span class="sc">+</span> Tenericutes <span class="sc">+</span> Verrucomicrobia <span class="sc">+</span> Fusobacteria) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Cyanobacteria, <span class="sc">-</span>Euryarchaeota, <span class="sc">-</span>Tenericutes, <span class="sc">-</span>Verrucomicrobia, <span class="sc">-</span>Fusobacteria)</span>
<span id="cb35-12"><a href="free-space.html#cb35-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-13"><a href="free-space.html#cb35-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Performing the HMP test</span></span>
<span id="cb35-14"><a href="free-space.html#cb35-14" aria-hidden="true" tabindex="-1"></a>group_data <span class="ot">&lt;-</span> <span class="fu">list</span>(control_otu, cf_otu)</span>
<span id="cb35-15"><a href="free-space.html#cb35-15" aria-hidden="true" tabindex="-1"></a>(xdc <span class="ot">&lt;-</span> HMP<span class="sc">::</span><span class="fu">Xdc.sevsample</span>(group_data))</span></code></pre></div>
<pre><code>## $`Xdc statistics`
## [1] -0.3510664
## 
## $`p value`
## [1] 1</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="free-space.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(<span class="fl">0.2769004</span>, <span class="dv">5</span>) <span class="co"># Looking at the Chi-Squared distribution</span></span></code></pre></div>
<pre><code>## [1] 0.9980551</code></pre>
<p>Since the P-value of the HMP test is nearly 1, we cannot reject the null hypothesis that there is no statistical significant distribution of phyla between the two groups.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="free-space.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting the OTU table</span></span>
<span id="cb39-2"><a href="free-space.html#cb39-2" aria-hidden="true" tabindex="-1"></a>ps_rel_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_rel_abund))</span>
<span id="cb39-3"><a href="free-space.html#cb39-3" aria-hidden="true" tabindex="-1"></a>ps_rel_otu <span class="ot">&lt;-</span> <span class="fu">t</span>(ps_rel_otu)</span>
<span id="cb39-4"><a href="free-space.html#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing the Bray-Curtis dissimilarity</span></span>
<span id="cb39-5"><a href="free-space.html#cb39-5" aria-hidden="true" tabindex="-1"></a>bc_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(ps_rel_otu, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb39-6"><a href="free-space.html#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Generating a matrix with the dissimilarities</span></span>
<span id="cb39-7"><a href="free-space.html#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="fu">as.matrix</span>(bc_dist)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##            ERR1331793 ERR1331872 ERR1331819 ERR1331794 ERR1331851
## ERR1331793  0.0000000  0.8801040  0.5975550  0.9767218  0.8684629
## ERR1331872  0.8801040  0.0000000  0.7590766  0.9596181  0.9206484
## ERR1331819  0.5975550  0.7590766  0.0000000  0.9556656  0.7810736
## ERR1331794  0.9767218  0.9596181  0.9556656  0.0000000  0.9693291
## ERR1331851  0.8684629  0.9206484  0.7810736  0.9693291  0.0000000</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="free-space.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving it as a dendrogram</span></span>
<span id="cb41-2"><a href="free-space.html#cb41-2" aria-hidden="true" tabindex="-1"></a>ward <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(<span class="fu">hclust</span>(bc_dist, <span class="at">method =</span> <span class="st">&quot;ward.D2&quot;</span>))</span>
<span id="cb41-3"><a href="free-space.html#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding color coding</span></span>
<span id="cb41-4"><a href="free-space.html#cb41-4" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_rel_abund))</span>
<span id="cb41-5"><a href="free-space.html#cb41-5" aria-hidden="true" tabindex="-1"></a>colorCode <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Control =</span> <span class="st">&quot;red&quot;</span>, <span class="st">`</span><span class="at">Chronic Fatigue</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb41-6"><a href="free-space.html#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="fu">labels_colors</span>(ward) <span class="ot">&lt;-</span> colorCode[meta<span class="sc">$</span>Status][<span class="fu">order.dendrogram</span>(ward)]</span>
<span id="cb41-7"><a href="free-space.html#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the dendrogram</span></span>
<span id="cb41-8"><a href="free-space.html#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ward)</span></code></pre></div>
<p><img src="_main_files/figure-html/looking%20at%20hierarchical%20clustering-1.png" width="672" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="free-space.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_heatmap</span>(cf)</span>
<span id="cb42-2"><a href="free-space.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_heatmap</span>(controls)</span></code></pre></div>
<p><strong>Looking at the alpha diversity</strong>
<br>
Alpha diversity looks at the diversity <em>within</em> a sample and how the observed OTU’s are distributed.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cv.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="using-github-and-zotero.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
