<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Free space | Portfolio Laurine Seelt</title>
  <meta name="description" content="This is the bookdown for my DSFB2 portfolio" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Free space | Portfolio Laurine Seelt" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the bookdown for my DSFB2 portfolio" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Free space | Portfolio Laurine Seelt" />
  
  <meta name="twitter:description" content="This is the bookdown for my DSFB2 portfolio" />
  

<meta name="author" content="Laurine Seelt" />


<meta name="date" content="2022-05-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cv.html"/>
<link rel="next" href="using-github-and-zotero.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="investigating-a-c.-elegans-plate-experiment.html"><a href="investigating-a-c.-elegans-plate-experiment.html"><i class="fa fa-check"></i><b>2</b> Investigating a <em>C. Elegans</em> plate experiment</a></li>
<li class="chapter" data-level="3" data-path="reading-two-different-studies-and-recreating-some-of-the-code.html"><a href="reading-two-different-studies-and-recreating-some-of-the-code.html"><i class="fa fa-check"></i><b>3</b> Reading two different studies and recreating some of the code</a>
<ul>
<li class="chapter" data-level="3.1" data-path="reading-two-different-studies-and-recreating-some-of-the-code.html"><a href="reading-two-different-studies-and-recreating-some-of-the-code.html#part-1"><i class="fa fa-check"></i><b>3.1</b> Part 1</a></li>
<li class="chapter" data-level="3.2" data-path="reading-two-different-studies-and-recreating-some-of-the-code.html"><a href="reading-two-different-studies-and-recreating-some-of-the-code.html#part-2"><i class="fa fa-check"></i><b>3.2</b> Part 2</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cleaning-up-according-to-the-guerrilla-analytics-framework.html"><a href="cleaning-up-according-to-the-guerrilla-analytics-framework.html"><i class="fa fa-check"></i><b>4</b> Cleaning up according to the Guerrilla analytics framework</a></li>
<li class="chapter" data-level="5" data-path="cv.html"><a href="cv.html"><i class="fa fa-check"></i><b>5</b> CV</a>
<ul>
<li class="chapter" data-level="5.1" data-path="cv.html"><a href="cv.html#persoonlijke-informatie"><i class="fa fa-check"></i><b>5.1</b> <strong>Persoonlijke informatie</strong></a></li>
<li class="chapter" data-level="5.2" data-path="cv.html"><a href="cv.html#opleiding"><i class="fa fa-check"></i><b>5.2</b> <strong>Opleiding</strong></a></li>
<li class="chapter" data-level="5.3" data-path="cv.html"><a href="cv.html#ervaringskills"><i class="fa fa-check"></i><b>5.3</b> <strong>Ervaring/skills</strong></a></li>
<li class="chapter" data-level="5.4" data-path="cv.html"><a href="cv.html#talen"><i class="fa fa-check"></i><b>5.4</b> <strong>Talen</strong></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="free-space.html"><a href="free-space.html"><i class="fa fa-check"></i><b>6</b> Free space</a>
<ul>
<li class="chapter" data-level="6.1" data-path="free-space.html"><a href="free-space.html#course-introduction-to-the-statistical-analysis-of-microbiome-data-in-r"><i class="fa fa-check"></i><b>6.1</b> Course: Introduction to the statistical analysis of microbiome data in R</a></li>
<li class="chapter" data-level="6.2" data-path="free-space.html"><a href="free-space.html#course-microbiota-analysis-in-r"><i class="fa fa-check"></i><b>6.2</b> Course: Microbiota analysis in R</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="using-github-and-zotero.html"><a href="using-github-and-zotero.html"><i class="fa fa-check"></i><b>7</b> Using GitHub and Zotero</a></li>
<li class="chapter" data-level="8" data-path="making-a-website-on-github.html"><a href="making-a-website-on-github.html"><i class="fa fa-check"></i><b>8</b> Making a website on GitHub</a></li>
<li class="chapter" data-level="9" data-path="working-with-dbeaver-and-sql.html"><a href="working-with-dbeaver-and-sql.html"><i class="fa fa-check"></i><b>9</b> Working with DBeaver and SQL</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio Laurine Seelt</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="free-space" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Free space<a href="free-space.html#free-space" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This was the fifth assignment for our portfolio. We had/have to fill 32 hours learning something new that will be useful for our internship next year and possibly our job.</p>
<ul>
<li>Uitzoeken welke automatiseringen er op microbiologische labs worden gebruikt.</li>
<li>Metagenomics?
<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03933-4" class="uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03933-4</a> (megaR package)
<a href="http://web.evolbio.mpg.de/~wang/Site/R_tutorial_files/16S%20Metagenomic%20Analysis%20Tutorial.pdf" class="uri">http://web.evolbio.mpg.de/~wang/Site/R_tutorial_files/16S%20Metagenomic%20Analysis%20Tutorial.pdf</a> (random tutorial)
<a href="https://www.mcs.anl.gov/~braithwaite/matR/docs-and-pubs/matR-user-manual.pdf" class="uri">https://www.mcs.anl.gov/~braithwaite/matR/docs-and-pubs/matR-user-manual.pdf</a> (matR package)
<a href="https://rpubs.com/jrandall7/EICC16s" class="uri">https://rpubs.com/jrandall7/EICC16s</a> (phyloseq)
<a href="https://cran.r-project.org/web/packages/microbial/microbial.pdf" class="uri">https://cran.r-project.org/web/packages/microbial/microbial.pdf</a> (microbial package)
<a href="https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html" class="uri">https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html</a> (aritkel over microbioom analyse workflow)</li>
</ul>
<p><a href="https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/" class="uri">https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/</a> (soort course over de analyse van microbioom data)
<a href="https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html" class="uri">https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html</a> (ook een course over de analyse van microbiota data)
<a href="https://grunwaldlab.github.io/analysis_of_microbiome_community_data_in_r/04--manipulating.html" class="uri">https://grunwaldlab.github.io/analysis_of_microbiome_community_data_in_r/04--manipulating.html</a> (nog een course)
- Python?</p>
<div id="course-introduction-to-the-statistical-analysis-of-microbiome-data-in-r" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Course: Introduction to the statistical analysis of microbiome data in R<a href="free-space.html#course-introduction-to-the-statistical-analysis-of-microbiome-data-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Link to the course: <a href="https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/" class="uri">https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/</a></p>
<p>In this course, we examined the differences in microbiota between patients with and without chronic fatigue syndrome.
The aspects that were looked at were: <br>
Taxonomic relative abundance <br>
Hierarchal clustering <br>
Alpha-diversity (microbiome diversity in just one sample) <br>
Beta-divesity (microbiome diversity in two or more samples) <br>
Diffrential abundance testing <br>
Predicting class labels <br></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="free-space.html#cb14-1" aria-hidden="true" tabindex="-1"></a>.cran_packages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;cowplot&quot;</span>, <span class="st">&quot;picante&quot;</span>, <span class="st">&quot;vegan&quot;</span>, <span class="st">&quot;HMP&quot;</span>, <span class="st">&quot;dendextend&quot;</span>, <span class="st">&quot;rms&quot;</span>, <span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb14-2"><a href="free-space.html#cb14-2" aria-hidden="true" tabindex="-1"></a>.bioc_packages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;phyloseq&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>, <span class="st">&quot;microbiome&quot;</span>, <span class="st">&quot;metagenomeSeq&quot;</span>, <span class="st">&quot;ALDEx2&quot;</span>)</span>
<span id="cb14-3"><a href="free-space.html#cb14-3" aria-hidden="true" tabindex="-1"></a>.inst <span class="ot">&lt;-</span> .cran_packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()</span>
<span id="cb14-4"><a href="free-space.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="sc">!</span>.inst)) {</span>
<span id="cb14-5"><a href="free-space.html#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(.cran_packages[<span class="sc">!</span>.inst])</span>
<span id="cb14-6"><a href="free-space.html#cb14-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-7"><a href="free-space.html#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb14-8"><a href="free-space.html#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb14-9"><a href="free-space.html#cb14-9" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(.bioc_packages, <span class="at">version =</span> <span class="st">&quot;3.15&quot;</span>) <span class="co">#Changed 3.9 to 3.15</span></span>
<span id="cb14-10"><a href="free-space.html#cb14-10" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;adw96/breakaway&quot;</span>)</span>
<span id="cb14-11"><a href="free-space.html#cb14-11" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="at">repo =</span> <span class="st">&quot;malucalle/selbal&quot;</span>) <span class="co">#changed UVic-omics to malucalle</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="free-space.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-2"><a href="free-space.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb15-3"><a href="free-space.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb15-4"><a href="free-space.html#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb15-5"><a href="free-space.html#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb15-6"><a href="free-space.html#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(picante)</span>
<span id="cb15-7"><a href="free-space.html#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ALDEx2)</span>
<span id="cb15-8"><a href="free-space.html#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(metagenomeSeq)</span>
<span id="cb15-9"><a href="free-space.html#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(HMP)</span>
<span id="cb15-10"><a href="free-space.html#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dendextend)</span>
<span id="cb15-11"><a href="free-space.html#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(selbal)</span>
<span id="cb15-12"><a href="free-space.html#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rms)</span>
<span id="cb15-13"><a href="free-space.html#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(breakaway)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="free-space.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Reading in the data</span></span>
<span id="cb16-2"><a href="free-space.html#cb16-2" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;Data/Data_raw/ps_giloteaux_2016.rds&quot;</span>)</span>
<span id="cb16-3"><a href="free-space.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Sorting the samples on total read count</span></span>
<span id="cb16-4"><a href="free-space.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(phyloseq<span class="sc">::</span><span class="fu">sample_sums</span>(ps))</span>
<span id="cb16-5"><a href="free-space.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing (samples with?) less than 5000 reads</span></span>
<span id="cb16-6"><a href="free-space.html#cb16-6" aria-hidden="true" tabindex="-1"></a>(ps <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(ps, phyloseq<span class="sc">::</span><span class="fu">sample_sums</span>(ps) <span class="sc">&gt;</span> <span class="dv">5000</span>))</span>
<span id="cb16-7"><a href="free-space.html#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing the OTU&#39;s that were only present in the removed samples</span></span>
<span id="cb16-8"><a href="free-space.html#cb16-8" aria-hidden="true" tabindex="-1"></a>(ps <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">prune_taxa</span>(phyloseq<span class="sc">::</span><span class="fu">taxa_sums</span>(ps) <span class="sc">&gt;</span> <span class="dv">0</span>, ps))</span>
<span id="cb16-9"><a href="free-space.html#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Assigning a new metadata field</span></span>
<span id="cb16-10"><a href="free-space.html#cb16-10" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Subject <span class="sc">==</span> <span class="st">&quot;Patient&quot;</span>, <span class="st">&quot;Chronic Fatique&quot;</span>, <span class="st">&quot;Control&quot;</span>)</span>
<span id="cb16-11"><a href="free-space.html#cb16-11" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Status <span class="ot">&lt;-</span> <span class="fu">factor</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps)<span class="sc">$</span>Status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>, <span class="st">&quot;Chronic Fatique&quot;</span>))</span>
<span id="cb16-12"><a href="free-space.html#cb16-12" aria-hidden="true" tabindex="-1"></a>ps <span class="sc">%&gt;%</span> sample_data <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(Status)</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="free-space.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Counting all the phyla in the dataset</span></span>
<span id="cb17-2"><a href="free-space.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(ps)[, <span class="st">&quot;Phylum&quot;</span>])</span>
<span id="cb17-3"><a href="free-space.html#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the samples to relative abundances</span></span>
<span id="cb17-4"><a href="free-space.html#cb17-4" aria-hidden="true" tabindex="-1"></a>ps_rel_abund <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">transform_sample_counts</span>(ps, <span class="cf">function</span>(x){x <span class="sc">/</span> <span class="fu">sum</span>(x)})</span>
<span id="cb17-5"><a href="free-space.html#cb17-5" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co"># Before we converted it to relative abundances</span></span>
<span id="cb17-6"><a href="free-space.html#cb17-6" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_rel_abund)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co"># After we calculated the relative abundances</span></span>
<span id="cb17-7"><a href="free-space.html#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="free-space.html#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the relative abundances, devided by the two groups</span></span>
<span id="cb17-9"><a href="free-space.html#cb17-9" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">plot_bar</span>(ps_rel_abund, <span class="at">fill =</span> <span class="st">&quot;Phylum&quot;</span>)<span class="sc">+</span></span>
<span id="cb17-10"><a href="free-space.html#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">color =</span> Phylum, <span class="at">fill =</span> Phylum), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>)<span class="sc">+</span></span>
<span id="cb17-11"><a href="free-space.html#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Relative Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>)<span class="sc">+</span></span>
<span id="cb17-12"><a href="free-space.html#cb17-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Status, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb17-13"><a href="free-space.html#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-14"><a href="free-space.html#cb17-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-15"><a href="free-space.html#cb17-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="free-space.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sorting the taxa by their relative abundance</span></span>
<span id="cb18-2"><a href="free-space.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(ps_rel_abund<span class="sc">@</span>tax_table[[<span class="st">&quot;Phylum&quot;</span>]], <span class="at">decreasing =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span>  phyloseq<span class="sc">::</span><span class="fu">plot_bar</span>(<span class="at">fill =</span> <span class="st">&quot;Phylum&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-3"><a href="free-space.html#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">color =</span> Phylum, <span class="at">fill =</span> Phylum), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">position =</span> <span class="st">&quot;stack&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-4"><a href="free-space.html#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Relative Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-5"><a href="free-space.html#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Status, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-6"><a href="free-space.html#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-7"><a href="free-space.html#cb18-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-8"><a href="free-space.html#cb18-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="free-space.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Grouping the taxa according to phylum and renaming the groups</span></span>
<span id="cb19-2"><a href="free-space.html#cb19-2" aria-hidden="true" tabindex="-1"></a>ps_phylum <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">tax_glom</span>(ps, <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb19-3"><a href="free-space.html#cb19-3" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">taxa_names</span>(ps_phylum) <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(ps_phylum)[, <span class="st">&quot;Phylum&quot;</span>]</span>
<span id="cb19-4"><a href="free-space.html#cb19-4" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_phylum)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb19-5"><a href="free-space.html#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="free-space.html#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Melting the data and making the boxplots</span></span>
<span id="cb19-7"><a href="free-space.html#cb19-7" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">psmelt</span>(ps_phylum) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="free-space.html#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data =</span> ., <span class="fu">aes</span>(<span class="at">x =</span> Status, <span class="at">y =</span> Abundance))<span class="sc">+</span></span>
<span id="cb19-9"><a href="free-space.html#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span></span>
<span id="cb19-10"><a href="free-space.html#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> OTU), <span class="at">height =</span> <span class="dv">0</span>, <span class="at">width =</span> <span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb19-11"><a href="free-space.html#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>)<span class="sc">+</span></span>
<span id="cb19-12"><a href="free-space.html#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> OTU, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="free-space.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making two subsets</span></span>
<span id="cb20-2"><a href="free-space.html#cb20-2" aria-hidden="true" tabindex="-1"></a>controls <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(ps_phylum, Status <span class="sc">==</span> <span class="st">&quot;Control&quot;</span>)</span>
<span id="cb20-3"><a href="free-space.html#cb20-3" aria-hidden="true" tabindex="-1"></a>cf <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">subset_samples</span>(ps_phylum, <span class="at">Status =</span> <span class="st">&quot;Chronic Fatigue&quot;</span>)</span>
<span id="cb20-4"><a href="free-space.html#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="free-space.html#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Looking at the OTU tables</span></span>
<span id="cb20-6"><a href="free-space.html#cb20-6" aria-hidden="true" tabindex="-1"></a>control_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(controls))</span>
<span id="cb20-7"><a href="free-space.html#cb20-7" aria-hidden="true" tabindex="-1"></a>cf_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(cf))</span>
<span id="cb20-8"><a href="free-space.html#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="free-space.html#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Grouping the rare phyla to improve testing</span></span>
<span id="cb20-10"><a href="free-space.html#cb20-10" aria-hidden="true" tabindex="-1"></a>control_otu <span class="ot">&lt;-</span> control_otu <span class="sc">%&gt;%</span> <span class="fu">t</span>(.) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Other =</span> Cyanobacteria <span class="sc">+</span> Euryarchaeota <span class="sc">+</span> Tenericutes <span class="sc">+</span> Verrucomicrobia <span class="sc">+</span> Fusobacteria) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Cyanobacteria, <span class="sc">-</span>Euryarchaeota, <span class="sc">-</span>Tenericutes, <span class="sc">-</span>Verrucomicrobia, <span class="sc">-</span>Fusobacteria)</span>
<span id="cb20-11"><a href="free-space.html#cb20-11" aria-hidden="true" tabindex="-1"></a>cf_otu <span class="ot">&lt;-</span> cf_otu <span class="sc">%&gt;%</span> <span class="fu">t</span>(.) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Other =</span> Cyanobacteria <span class="sc">+</span> Euryarchaeota <span class="sc">+</span> Tenericutes <span class="sc">+</span> Verrucomicrobia <span class="sc">+</span> Fusobacteria) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Cyanobacteria, <span class="sc">-</span>Euryarchaeota, <span class="sc">-</span>Tenericutes, <span class="sc">-</span>Verrucomicrobia, <span class="sc">-</span>Fusobacteria)</span>
<span id="cb20-12"><a href="free-space.html#cb20-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-13"><a href="free-space.html#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Performing the HMP test</span></span>
<span id="cb20-14"><a href="free-space.html#cb20-14" aria-hidden="true" tabindex="-1"></a>group_data <span class="ot">&lt;-</span> <span class="fu">list</span>(control_otu, cf_otu)</span>
<span id="cb20-15"><a href="free-space.html#cb20-15" aria-hidden="true" tabindex="-1"></a>(xdc <span class="ot">&lt;-</span> HMP<span class="sc">::</span><span class="fu">Xdc.sevsample</span>(group_data))</span>
<span id="cb20-16"><a href="free-space.html#cb20-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-17"><a href="free-space.html#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(<span class="fl">0.2769004</span>, <span class="dv">5</span>) <span class="co"># Looking at the Chi-Squared distribution</span></span></code></pre></div>
<p>Since the P-value of the HMP test is nearly 1, we cannot reject the null hypothesis that there is no statistical significant distribution of phyla between the two groups.</p>
<p><strong>Looking at the hierarchical clustering in the samples</strong>
<br></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="free-space.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting the OTU table</span></span>
<span id="cb21-2"><a href="free-space.html#cb21-2" aria-hidden="true" tabindex="-1"></a>ps_rel_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_rel_abund))</span>
<span id="cb21-3"><a href="free-space.html#cb21-3" aria-hidden="true" tabindex="-1"></a>ps_rel_otu <span class="ot">&lt;-</span> <span class="fu">t</span>(ps_rel_otu)</span>
<span id="cb21-4"><a href="free-space.html#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing the Bray-Curtis dissimilarity</span></span>
<span id="cb21-5"><a href="free-space.html#cb21-5" aria-hidden="true" tabindex="-1"></a>bc_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(ps_rel_otu, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb21-6"><a href="free-space.html#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Generating a matrix with the dissimilarities</span></span>
<span id="cb21-7"><a href="free-space.html#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="fu">as.matrix</span>(bc_dist)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb21-8"><a href="free-space.html#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving it as a dendrogram</span></span>
<span id="cb21-9"><a href="free-space.html#cb21-9" aria-hidden="true" tabindex="-1"></a>ward <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(<span class="fu">hclust</span>(bc_dist, <span class="at">method =</span> <span class="st">&quot;ward.D2&quot;</span>))</span>
<span id="cb21-10"><a href="free-space.html#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding color coding</span></span>
<span id="cb21-11"><a href="free-space.html#cb21-11" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_rel_abund))</span>
<span id="cb21-12"><a href="free-space.html#cb21-12" aria-hidden="true" tabindex="-1"></a>colorCode <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Control =</span> <span class="st">&quot;red&quot;</span>, <span class="st">`</span><span class="at">Chronic Fatigue</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb21-13"><a href="free-space.html#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="fu">labels_colors</span>(ward) <span class="ot">&lt;-</span> colorCode[meta<span class="sc">$</span>Status][<span class="fu">order.dendrogram</span>(ward)]</span>
<span id="cb21-14"><a href="free-space.html#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the dendrogram</span></span>
<span id="cb21-15"><a href="free-space.html#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ward)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="free-space.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_heatmap</span>(cf)</span>
<span id="cb22-2"><a href="free-space.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_heatmap</span>(controls)</span></code></pre></div>
<p><strong>Looking at the alpha diversity</strong>
<br>
Alpha diversity looks at the diversity <em>within</em> a sample and how the observed OTU’s are distributed.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="free-space.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the observed richness</span></span>
<span id="cb23-2"><a href="free-space.html#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">data.frame</span>(<span class="st">&quot;total_reads&quot;</span> <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">sample_sums</span>(ps),</span>
<span id="cb23-3"><a href="free-space.html#cb23-3" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;observed&quot;</span> <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">estimate_richness</span>(ps, <span class="at">measures =</span> <span class="st">&quot;Observed&quot;</span>)[, <span class="dv">1</span>]),</span>
<span id="cb23-4"><a href="free-space.html#cb23-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> total_reads, <span class="at">y =</span> observed)) <span class="sc">+</span></span>
<span id="cb23-5"><a href="free-space.html#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb23-6"><a href="free-space.html#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb23-7"><a href="free-space.html#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Total Reads&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Observed richness</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<p>Now we’re going to subsample the reads, plot them and test for group differences
<br></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="free-space.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subsample reads</span></span>
<span id="cb24-2"><a href="free-space.html#cb24-2" aria-hidden="true" tabindex="-1"></a>(ps_rare <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">rarefy_even_depth</span>(ps, <span class="at">rngseed =</span> <span class="dv">123</span>, <span class="at">replace =</span> <span class="cn">FALSE</span>))</span>
<span id="cb24-3"><a href="free-space.html#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(phyloseq<span class="sc">::</span><span class="fu">sample_sums</span>(ps_rare))</span>
<span id="cb24-4"><a href="free-space.html#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a dataframe with the adiv measures, another package to analyse and measure biodiversity</span></span>
<span id="cb24-5"><a href="free-space.html#cb24-5" aria-hidden="true" tabindex="-1"></a>adiv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&quot;Observed&quot;</span> <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">estimate_richness</span>(ps_rare, <span class="at">measures =</span> <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb24-6"><a href="free-space.html#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Shannon&quot;</span> <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">estimate_richness</span>(ps_rare, <span class="at">measures =</span> <span class="st">&quot;Shannon&quot;</span>),</span>
<span id="cb24-7"><a href="free-space.html#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;PD&quot;</span> <span class="ot">=</span> picante<span class="sc">::</span><span class="fu">pd</span>(<span class="at">samp =</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">data.frame</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_rare)))), <span class="at">include.root =</span> <span class="cn">FALSE</span>, <span class="at">tree=</span>phyloseq<span class="sc">::</span><span class="fu">phy_tree</span>(ps_rare))[, <span class="dv">1</span>],</span>
<span id="cb24-8"><a href="free-space.html#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Status&quot;</span> <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_rare)<span class="sc">$</span>Status) <span class="co"># Zelf &quot;include.root = FALSE&quot; toegevoegd, anders kreeg ik steeds een error</span></span>
<span id="cb24-9"><a href="free-space.html#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(adiv)</span>
<span id="cb24-10"><a href="free-space.html#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the adiv measures</span></span>
<span id="cb24-11"><a href="free-space.html#cb24-11" aria-hidden="true" tabindex="-1"></a>adiv <span class="sc">%&gt;%</span> <span class="fu">gather</span>(<span class="at">key =</span> metric, <span class="at">value =</span> value, <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>, <span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;PD&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-12"><a href="free-space.html#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">factor</span>(metric, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Observed&quot;</span>, <span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;PD&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb24-13"><a href="free-space.html#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Status, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb24-14"><a href="free-space.html#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.color =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb24-15"><a href="free-space.html#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> Status), <span class="at">height =</span> <span class="dv">0</span>, <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb24-16"><a href="free-space.html#cb24-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb24-17"><a href="free-space.html#cb24-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>metric, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb24-18"><a href="free-space.html#cb24-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb24-19"><a href="free-space.html#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarising the data</span></span>
<span id="cb24-20"><a href="free-space.html#cb24-20" aria-hidden="true" tabindex="-1"></a>adiv <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Status) <span class="sc">%&gt;%</span> </span>
<span id="cb24-21"><a href="free-space.html#cb24-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">median_observed =</span> <span class="fu">median</span>(Observed),</span>
<span id="cb24-22"><a href="free-space.html#cb24-22" aria-hidden="true" tabindex="-1"></a>                   <span class="at">median_shannon =</span> <span class="fu">median</span>(Shannon),</span>
<span id="cb24-23"><a href="free-space.html#cb24-23" aria-hidden="true" tabindex="-1"></a>                   <span class="at">median_pd =</span> <span class="fu">median</span>(PD))</span>
<span id="cb24-24"><a href="free-space.html#cb24-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Performing the Wilcoxon tests for the three estimates</span></span>
<span id="cb24-25"><a href="free-space.html#cb24-25" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(Observed <span class="sc">~</span> Status, <span class="at">data =</span> adiv, <span class="at">exact =</span> <span class="cn">FALSE</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-26"><a href="free-space.html#cb24-26" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(Shannon <span class="sc">~</span> Status, <span class="at">data =</span> adiv, <span class="at">conf.int =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-27"><a href="free-space.html#cb24-27" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(PD <span class="sc">~</span> Status, <span class="at">data =</span> adiv, <span class="at">conf.int =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><strong>Looking at the beta diversity</strong>
<br>
Beta diversity looks at the diversity <em>between</em> samples and how it’s similar or dissimilar.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="free-space.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transforming the Total Read Counts according to the Centered Log Ratio method</span></span>
<span id="cb25-2"><a href="free-space.html#cb25-2" aria-hidden="true" tabindex="-1"></a>(ps_clr <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">transform</span>(ps, <span class="st">&quot;clr&quot;</span>))</span>
<span id="cb25-3"><a href="free-space.html#cb25-3" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co"># Old OTU table</span></span>
<span id="cb25-4"><a href="free-space.html#cb25-4" aria-hidden="true" tabindex="-1"></a>phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps_clr)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co"># New OTU table</span></span>
<span id="cb25-5"><a href="free-space.html#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="co"># The value are now the dominance for each taxa relative to the geometric mean of all taxa on the logarithmic scale</span></span></code></pre></div>
<p><br>
<br></p>
</div>
<div id="course-microbiota-analysis-in-r" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Course: Microbiota analysis in R<a href="free-space.html#course-microbiota-analysis-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Link to the course: <a href="https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html" class="uri">https://rstudio-pubs-static.s3.amazonaws.com/268156_d3ea37937f4f4469839ab6fa2c483842.html</a>
<br>
<br>
In this course, we will be looking at the fecal bacterial microbiota of 8 calves at ages 2 weeks, 8 weeks and 1 year old and correlating them with variables such as weight gain, expressed as Average Daily Gain in Kilograms(ADGKG), and gastrointestinal short chain fatty acids, SCFA.
<br></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="free-space.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;ape&quot;)</span></span>
<span id="cb26-2"><a href="free-space.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb26-3"><a href="free-space.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb26-4"><a href="free-space.html#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb26-5"><a href="free-space.html#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb26-6"><a href="free-space.html#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gplots)</span>
<span id="cb26-7"><a href="free-space.html#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lme4)</span>
<span id="cb26-8"><a href="free-space.html#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phangorn)</span>
<span id="cb26-9"><a href="free-space.html#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb26-10"><a href="free-space.html#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb26-11"><a href="free-space.html#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb26-12"><a href="free-space.html#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb26-13"><a href="free-space.html#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;VennDiagram&quot;)</span></span>
<span id="cb26-14"><a href="free-space.html#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VennDiagram)</span>
<span id="cb26-15"><a href="free-space.html#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Installed Java from: https://www.java.com/en/download/manual.jsp</span></span>
<span id="cb26-16"><a href="free-space.html#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Installed Java JDk from: https://www.oracle.com/java/technologies/downloads/#jdk18-windows</span></span>
<span id="cb26-17"><a href="free-space.html#cb26-17" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;rJava&quot;)</span></span>
<span id="cb26-18"><a href="free-space.html#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rJava)</span>
<span id="cb26-19"><a href="free-space.html#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;venneuler&quot;)</span></span>
<span id="cb26-20"><a href="free-space.html#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="co"># library(venneuler) # R kapt er steeds mee als ik deze wil laden</span></span>
<span id="cb26-21"><a href="free-space.html#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb26-22"><a href="free-space.html#cb26-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb26-23"><a href="free-space.html#cb26-23" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install(version = &quot;3.15&quot;) # Instead of source(&quot;https://bioconductor.org/biocLite.R&quot;)</span></span>
<span id="cb26-24"><a href="free-space.html#cb26-24" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install(&quot;phyloseq&quot;) # Instead of biocLite(&quot;phyloseq&quot;) ?</span></span>
<span id="cb26-25"><a href="free-space.html#cb26-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="free-space.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Downloaded the data from: https://github.com/kdillmcfarland/workshops_UW_Madison/tree/master/Microbiota_analysis_R/Data</span></span>
<span id="cb27-2"><a href="free-space.html#cb27-2" aria-hidden="true" tabindex="-1"></a>OTU <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;Data/example.final.an.unique_list.0.03.norm.shared.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb27-3"><a href="free-space.html#cb27-3" aria-hidden="true" tabindex="-1"></a>tax <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;Data/example.final.an.unique_list.0.03.cons.taxonomy.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb27-4"><a href="free-space.html#cb27-4" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;Data/example.metadata.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb27-5"><a href="free-space.html#cb27-5" aria-hidden="true" tabindex="-1"></a>SCFA <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;Data/example.SCFA.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="free-space.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting the &quot;Group&quot; column as row names in the OTU dataset</span></span>
<span id="cb28-2"><a href="free-space.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(OTU) <span class="ot">=</span> OTU<span class="sc">$</span>Group</span>
<span id="cb28-3"><a href="free-space.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing the columns that are not OTU counts</span></span>
<span id="cb28-4"><a href="free-space.html#cb28-4" aria-hidden="true" tabindex="-1"></a>OTU.clean <span class="ot">=</span> OTU[,<span class="sc">-</span><span class="fu">which</span>(<span class="fu">names</span>(OTU) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;label&quot;</span>, <span class="st">&quot;numOtus&quot;</span>, <span class="st">&quot;Group&quot;</span>))]</span>
<span id="cb28-5"><a href="free-space.html#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Setting the &quot;OTU&quot; column as row names in the taxonomy table</span></span>
<span id="cb28-6"><a href="free-space.html#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(tax) <span class="ot">=</span> tax<span class="sc">$</span>OTU</span>
<span id="cb28-7"><a href="free-space.html#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing the OTU&#39;s that aren&#39;t present in the OTU.clean dataset</span></span>
<span id="cb28-8"><a href="free-space.html#cb28-8" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">=</span> tax[<span class="fu">row.names</span>(tax) <span class="sc">%in%</span> <span class="fu">colnames</span>(OTU.clean),]</span>
<span id="cb28-9"><a href="free-space.html#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Separating the taxonomy table so each level has its own column</span></span>
<span id="cb28-10"><a href="free-space.html#cb28-10" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">=</span> <span class="fu">separate</span>(tax.clean, Taxonomy, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;Domain&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>, <span class="st">&quot;Strain&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb28-11"><a href="free-space.html#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing the &quot;Size&quot;, &quot;Strain&quot; and &quot;OTU&quot; columns because these are now row names</span></span>
<span id="cb28-12"><a href="free-space.html#cb28-12" aria-hidden="true" tabindex="-1"></a>tax.clean <span class="ot">=</span> tax.clean[,<span class="sc">-</span><span class="fu">which</span>(<span class="fu">names</span>(tax.clean) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Size&quot;</span>, <span class="st">&quot;Strain&quot;</span>, <span class="st">&quot;OTU&quot;</span>))]</span>
<span id="cb28-13"><a href="free-space.html#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Making sure the three datasets have samples in the same order</span></span>
<span id="cb28-14"><a href="free-space.html#cb28-14" aria-hidden="true" tabindex="-1"></a>OTU.clean <span class="ot">=</span> OTU.clean[<span class="fu">order</span>(<span class="fu">row.names</span>(OTU.clean)),]</span>
<span id="cb28-15"><a href="free-space.html#cb28-15" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">=</span> meta[<span class="fu">order</span>(<span class="fu">row.names</span>(meta)),]</span>
<span id="cb28-16"><a href="free-space.html#cb28-16" aria-hidden="true" tabindex="-1"></a>SCFA <span class="ot">=</span> SCFA[<span class="fu">order</span>(<span class="fu">row.names</span>(SCFA)),]</span>
<span id="cb28-17"><a href="free-space.html#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="co"># set.seed to make the analysis reproducible</span></span>
<span id="cb28-18"><a href="free-space.html#cb28-18" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">8765</span>)</span></code></pre></div>
<p><strong>Looking at the alpha diversity</strong> <br>
Alpha diversity is the diversity <em>within</em> a sample. It looks at richness, which is the <em>amount</em> of OTU’s in each sample, and at evenness, which is how evenly the different OTU’s are distributed within the sample. <br></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="free-space.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a 2x2 plot environment so we can see all 4 metrics at once</span></span>
<span id="cb29-2"><a href="free-space.html#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb29-3"><a href="free-space.html#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the four metrics</span></span>
<span id="cb29-4"><a href="free-space.html#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meta<span class="sc">$</span>shannon, <span class="at">main =</span> <span class="st">&quot;Shannon diversity&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">10</span>)</span>
<span id="cb29-5"><a href="free-space.html#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meta<span class="sc">$</span>simpson, <span class="at">main =</span> <span class="st">&quot;Simpson diversity&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">10</span>)</span>
<span id="cb29-6"><a href="free-space.html#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meta<span class="sc">$</span>chao, <span class="at">main =</span> <span class="st">&quot;Chao richness&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">15</span>)</span>
<span id="cb29-7"><a href="free-space.html#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meta<span class="sc">$</span>ace, <span class="at">main =</span> <span class="st">&quot;ACE richness&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/looking%20at%20the%20alpha%20diversity-1.png" width="672" />
<br>
None of the data are normally distributed. Simpson diversity is very often skewed as seen in this histogram, so we’ll calculate 1/Simpson and plot the metrics again.
<br></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="free-space.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a 2x2 environment</span></span>
<span id="cb30-2"><a href="free-space.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb30-3"><a href="free-space.html#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the four metrics again</span></span>
<span id="cb30-4"><a href="free-space.html#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meta<span class="sc">$</span>shannon, <span class="at">main =</span> <span class="st">&quot;Shannon diversity&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">10</span>)</span>
<span id="cb30-5"><a href="free-space.html#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="dv">1</span><span class="sc">/</span>meta<span class="sc">$</span>simpson, <span class="at">main =</span> <span class="st">&quot;Inverse Simpson diversity&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">10</span>)</span>
<span id="cb30-6"><a href="free-space.html#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meta<span class="sc">$</span>chao, <span class="at">main =</span> <span class="st">&quot;Chao richness&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">15</span>)</span>
<span id="cb30-7"><a href="free-space.html#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meta<span class="sc">$</span>ace, <span class="at">main =</span> <span class="st">&quot;ACE richness&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">breaks =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-10-1.png" width="672" />
<br>
Now the Simpson diversity is distributed similarly to the other richness metrics.
<br>
Next, we’ll test the four metrics for normal distribution.
<br></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="free-space.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(meta<span class="sc">$</span>shannon)</span></code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  meta$shannon
## W = 0.91511, p-value = 0.0456</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="free-space.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(<span class="dv">1</span><span class="sc">/</span>meta<span class="sc">$</span>simpson)</span></code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  1/meta$simpson
## W = 0.74821, p-value = 4.69e-05</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="free-space.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(meta<span class="sc">$</span>chao)</span></code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  meta$chao
## W = 0.80636, p-value = 0.0003749</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="free-space.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(meta<span class="sc">$</span>ace)</span></code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  meta$ace
## W = 0.83017, p-value = 0.0009573</code></pre>
<p><br>
None of the richness metrics are normally distributed, which was to be expected from the graphs we’ve seen. So we cannot run any tests that assume the data is normally distributed.
<br>
For illustration purposes, we’ll run the ANOVA test with the Shannon’s diversity because that’s the closest to normally distributed. We’ll look at if age impacts the Shannon diversity of the fecal microbiota.
<br></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="free-space.html#cb39-1" aria-hidden="true" tabindex="-1"></a>aov.shannon.age <span class="ot">=</span> <span class="fu">aov</span>(shannon <span class="sc">~</span> AgeGroup, <span class="at">data =</span> meta)</span>
<span id="cb39-2"><a href="free-space.html#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(aov.shannon.age)</span></code></pre></div>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## AgeGroup     2  42.98  21.489   103.4 1.35e-11 ***
## Residuals   21   4.36   0.208                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><br>
We’ll run Tukey’s honest significance test to do pairwise comparisons between groups and correct for multiple comparisons.
<br></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="free-space.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">TukeyHSD</span>(aov.shannon.age)</span></code></pre></div>
<pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
## 
## Fit: aov(formula = shannon ~ AgeGroup, data = meta)
## 
## $AgeGroup
##             diff        lwr       upr   p adj
## 2w-1yr -3.270063 -3.8446230 -2.695503 0.0e+00
## 8w-1yr -1.830903 -2.4054628 -1.256342 2.0e-07
## 8w-2w   1.439160  0.8646001  2.013720 8.5e-06</code></pre>
<p><br>
It’s clear that all age groups have significantly different diversity. In a plot, we can clearly see that <em>diversity</em> increases with ages.
<br></p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="free-space.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Re-ordering the groups</span></span>
<span id="cb43-2"><a href="free-space.html#cb43-2" aria-hidden="true" tabindex="-1"></a>meta<span class="sc">$</span>AgeGroup.ord <span class="ot">=</span> <span class="fu">factor</span>(meta<span class="sc">$</span>AgeGroup, <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>))</span>
<span id="cb43-3"><a href="free-space.html#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Returning the plot area to 1x1</span></span>
<span id="cb43-4"><a href="free-space.html#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb43-5"><a href="free-space.html#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the diversity</span></span>
<span id="cb43-6"><a href="free-space.html#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(shannon <span class="sc">~</span> AgeGroup.ord, <span class="at">data =</span> meta, </span>
<span id="cb43-7"><a href="free-space.html#cb43-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">&quot;Shannon&#39;s diversity&quot;</span>,</span>
<span id="cb43-8"><a href="free-space.html#cb43-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">&quot;Age group&quot;</span>) <span class="co"># Added the xlab myself because it looks better :)</span></span></code></pre></div>
<p><img src="_main_files/figure-html/plotting%20the%20diversity-1.png" width="672" />
<br>
To illustrate some non-parametric tests, we’ll use Chao’s richness estimate. Age is categorical, so we’ll use Kruskal-Wallis.
<br></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="free-space.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kruskal.test</span>(chao <span class="sc">~</span> AgeGroup, <span class="at">data =</span> meta)</span></code></pre></div>
<pre><code>## 
##  Kruskal-Wallis rank sum test
## 
## data:  chao by AgeGroup
## Kruskal-Wallis chi-squared = 19.28, df = 2, p-value = 6.507e-05</code></pre>
<p><br>
We can also test pairwise within age groups with Wilcoxon Rank Sum Tests.
<br></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="free-space.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(meta<span class="sc">$</span>chao, meta<span class="sc">$</span>AgeGroup, <span class="at">p.adjust.method =</span> <span class="st">&quot;fdr&quot;</span>)</span></code></pre></div>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum exact test 
## 
## data:  meta$chao and meta$AgeGroup 
## 
##    1yr     2w     
## 2w 0.00023 -      
## 8w 0.00023 0.00186
## 
## P value adjustment method: fdr</code></pre>
<p><br>
Just like the diversity, <strong>richness</strong> also increases with age.
<br></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="free-space.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a 1x1 plot environment</span></span>
<span id="cb48-2"><a href="free-space.html#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb48-3"><a href="free-space.html#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the richness</span></span>
<span id="cb48-4"><a href="free-space.html#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(chao <span class="sc">~</span> AgeGroup.ord, <span class="at">data =</span> meta,</span>
<span id="cb48-5"><a href="free-space.html#cb48-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">&quot;Chao richness&quot;</span>,</span>
<span id="cb48-6"><a href="free-space.html#cb48-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">&quot;Age group&quot;</span>) <span class="co"># Again, added the xlab myself for aesthetics :)</span></span></code></pre></div>
<p><img src="_main_files/figure-html/plotting%20the%20richness-1.png" width="672" />
<br>
Average Daily Gain is a continuous variable, so we’ll use a linear model to visualise it.
<br>
We’ll use Shannon’s diversity again to run some tests that are meant for normally distributed data. We’ll take a look at if the ADG impacts the Shannon diversity of the fecal microbiota.
<br></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="free-space.html#cb49-1" aria-hidden="true" tabindex="-1"></a>glm.shannon.ADG <span class="ot">=</span> <span class="fu">glm</span>(shannon <span class="sc">~</span> ADGKG, <span class="at">data =</span> meta)</span>
<span id="cb49-2"><a href="free-space.html#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm.shannon.ADG)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = shannon ~ ADGKG, data = meta)
## 
## Deviance Residuals: 
##      Min        1Q    Median        3Q       Max  
## -2.49110  -1.11216  -0.01749   1.53658   1.84728  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)   
## (Intercept)  3.62565    1.01390   3.576  0.00169 **
## ADGKG       -0.03407    0.97805  -0.035  0.97253   
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for gaussian family taken to be 2.151815)
## 
##     Null deviance: 47.343  on 23  degrees of freedom
## Residual deviance: 47.340  on 22  degrees of freedom
## AIC: 90.412
## 
## Number of Fisher Scoring iterations: 2</code></pre>
<p><br>
This shows that the intercept of our model is signficantly different from 0, bu the slope is not, and the slope is our variable of interest.
<br>
Next, let’s plot the glm.
<br></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="free-space.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(shannon <span class="sc">~</span> ADGKG, <span class="at">data =</span> meta)</span>
<span id="cb51-2"><a href="free-space.html#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(glm.shannon.ADG)</span></code></pre></div>
<p><img src="_main_files/figure-html/plotting%20the%20glm-1.png" width="672" />
<br>
To illustrate non-normally distributed data, we’ll use Chao’s richness estimate again. We’ll first use the Gaussian distribution, but we already know this isn’t a good fit.
<br></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="free-space.html#cb52-1" aria-hidden="true" tabindex="-1"></a>gaussian.chao.adg <span class="ot">=</span> <span class="fu">glm</span>(chao <span class="sc">~</span>ADGKG, <span class="at">data =</span> meta, <span class="at">family =</span> <span class="st">&quot;gaussian&quot;</span>)</span>
<span id="cb52-2"><a href="free-space.html#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb52-3"><a href="free-space.html#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(gaussian.chao.adg, <span class="at">which =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/plotting%20Gaussian-1.png" width="672" />
<br>
Next, we’ll plot the quasipoisson distribuiton.
<br></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="free-space.html#cb53-1" aria-hidden="true" tabindex="-1"></a>qp.chao.ADG <span class="ot">=</span> <span class="fu">glm</span>(chao <span class="sc">~</span> ADGKG, <span class="at">data =</span> meta, <span class="at">family =</span> <span class="st">&quot;quasipoisson&quot;</span>)</span>
<span id="cb53-2"><a href="free-space.html#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb53-3"><a href="free-space.html#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(qp.chao.ADG, <span class="at">which =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/plotting%20quasipoisson-1.png" width="672" />
<br>
The quasipoisson distribution fits much better than the Gaussian distribution, so we’ll use quasipoisson for further calculations.
<br></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="free-space.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(qp.chao.ADG)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = chao ~ ADGKG, family = &quot;quasipoisson&quot;, data = meta)
## 
## Deviance Residuals: 
##    Min      1Q  Median      3Q     Max  
## -24.36  -17.05  -10.66   18.81   26.91  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)   6.4528     0.5561  11.605 7.54e-11 ***
## ADGKG        -0.1859     0.5438  -0.342    0.736    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for quasipoisson family taken to be 374.2485)
## 
##     Null deviance: 8117.2  on 23  degrees of freedom
## Residual deviance: 8074.4  on 22  degrees of freedom
## AIC: NA
## 
## Number of Fisher Scoring iterations: 5</code></pre>
<p><br>
We’ll plot Chao and ADGKG again, to look at the correlation.
<br></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="free-space.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb56-2"><a href="free-space.html#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">log</span>(chao) <span class="sc">~</span> ADGKG, <span class="at">data =</span> meta, <span class="at">ylab =</span> <span class="st">&quot;ln(Chao&#39;s richness)&quot;</span>)</span>
<span id="cb56-3"><a href="free-space.html#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(qp.chao.ADG)</span></code></pre></div>
<p><img src="_main_files/figure-html/plotting%20Chao%20and%20ADGKG-1.png" width="672" />
<br>
Looking at the graph, there is nog significant correlation between Chao’s richness and the Average Daily Gain.
<br>
To test if age and ADG impacts diversity, we’ll conduct an ANOVA test.
<br></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="free-space.html#cb57-1" aria-hidden="true" tabindex="-1"></a>aov.shannon.all <span class="ot">=</span> <span class="fu">aov</span>(shannon <span class="sc">~</span> AgeGroup<span class="sc">*</span>ADGKG, <span class="at">data =</span> meta)</span>
<span id="cb57-2"><a href="free-space.html#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(aov.shannon.all)</span></code></pre></div>
<pre><code>##                Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## AgeGroup        2  42.98  21.489  95.472 2.61e-10 ***
## ADGKG           1   0.05   0.054   0.239    0.631    
## AgeGroup:ADGKG  2   0.26   0.130   0.576    0.572    
## Residuals      18   4.05   0.225                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><br>
We can see that the interaction between age and ADG does not significantly impact Shannon’s diversity, so we’ll remove that variable and run the test again.
<br></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="free-space.html#cb59-1" aria-hidden="true" tabindex="-1"></a>aov.shannon.all2 <span class="ot">=</span> <span class="fu">aov</span>(shannon <span class="sc">~</span> AgeGroup<span class="sc">+</span>ADGKG, <span class="at">data =</span> meta)</span>
<span id="cb59-2"><a href="free-space.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(aov.shannon.all2)</span></code></pre></div>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## AgeGroup     2  42.98  21.489   99.70 3.96e-11 ***
## ADGKG        1   0.05   0.054    0.25    0.623    
## Residuals   20   4.31   0.216                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><br>
This test tells us that only age impacts Shannon’s diversity significantly, but we do not know which age groups differ significantly from eachother. To test this, we can run TukeyHSD with just age.
<br></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="free-space.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">TukeyHSD</span>(aov.shannon.all)</span></code></pre></div>
<pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
## 
## Fit: aov(formula = shannon ~ AgeGroup * ADGKG, data = meta)
## 
## $AgeGroup
##             diff       lwr       upr    p adj
## 2w-1yr -3.270063 -3.875469 -2.664657 0.00e+00
## 8w-1yr -1.830903 -2.436309 -1.225496 1.20e-06
## 8w-2w   1.439160  0.833754  2.044567 2.81e-05</code></pre>
<p><br>
With this test, however, ADGKG is ignored because it’s continuous. <br>
We’ll run the glm instead.
<br></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="free-space.html#cb63-1" aria-hidden="true" tabindex="-1"></a>glm.shannon.all <span class="ot">=</span> <span class="fu">glm</span>(shannon <span class="sc">~</span> AgeGroup<span class="sc">*</span>ADGKG, <span class="at">data =</span> meta)</span>
<span id="cb63-2"><a href="free-space.html#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm.shannon.all)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = shannon ~ AgeGroup * ADGKG, data = meta)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -1.0301  -0.2468   0.0894   0.1572   0.7624  
## 
## Coefficients:
##                  Estimate Std. Error t value Pr(&gt;|t|)  
## (Intercept)        5.7123     2.5928   2.203   0.0409 *
## AgeGroup2w        -3.3969     2.6197  -1.297   0.2111  
## AgeGroup8w        -2.9610     2.7554  -1.075   0.2967  
## ADGKG             -0.4481     2.7599  -0.162   0.8728  
## AgeGroup2w:ADGKG   0.1228     2.7848   0.044   0.9653  
## AgeGroup8w:ADGKG   1.0750     2.8763   0.374   0.7130  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for gaussian family taken to be 0.22508)
## 
##     Null deviance: 47.3425  on 23  degrees of freedom
## Residual deviance:  4.0514  on 18  degrees of freedom
## AIC: 39.413
## 
## Number of Fisher Scoring iterations: 2</code></pre>
<p><br>
We can see that none of the interaction terms, which are AgeGroup2w:ADGKG and AgeGroup8w:ADGKG, are significant. So we’ll remove these, by changing AgeGroup*ADGKG to AgeGroup+ADGKG, and run the glm again.
<br></p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="free-space.html#cb65-1" aria-hidden="true" tabindex="-1"></a>glm.shannon.all2 <span class="ot">=</span> <span class="fu">glm</span>(shannon <span class="sc">~</span> AgeGroup<span class="sc">+</span>ADGKG, <span class="at">data =</span> meta)</span>
<span id="cb65-2"><a href="free-space.html#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm.shannon.all2)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = shannon ~ AgeGroup + ADGKG, data = meta)
## 
## Deviance Residuals: 
##      Min        1Q    Median        3Q       Max  
## -0.95299  -0.25858   0.07643   0.30409   0.74487  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)   5.4459     0.3487  15.619 1.14e-12 ***
## AgeGroup2w   -3.2760     0.2324 -14.094 7.55e-12 ***
## AgeGroup8w   -1.7989     0.2408  -7.471 3.30e-07 ***
## ADGKG        -0.1639     0.3281  -0.500    0.623    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for gaussian family taken to be 0.2155447)
## 
##     Null deviance: 47.3425  on 23  degrees of freedom
## Residual deviance:  4.3109  on 20  degrees of freedom
## AIC: 36.903
## 
## Number of Fisher Scoring iterations: 2</code></pre>
<p><br>
Now that we’ve removed the interaction terms, the glm model shows age as significant.
<br>
This can also be done with non-normally distributed data, like Chao’s richness.
<br></p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="free-space.html#cb67-1" aria-hidden="true" tabindex="-1"></a>qp.chao.all <span class="ot">=</span> <span class="fu">glm</span>(chao <span class="sc">~</span> AgeGroup<span class="sc">*</span>ADGKG, <span class="at">data =</span> meta, <span class="at">family =</span> <span class="st">&quot;quasipoisson&quot;</span>)</span>
<span id="cb67-2"><a href="free-space.html#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(qp.chao.all)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = chao ~ AgeGroup * ADGKG, family = &quot;quasipoisson&quot;, 
##     data = meta)
## 
## Deviance Residuals: 
##    Min      1Q  Median      3Q     Max  
## -7.774  -3.430  -0.140   3.692   5.277  
## 
## Coefficients:
##                  Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)       6.99825    0.71122   9.840 1.14e-08 ***
## AgeGroup2w       -1.61539    0.75272  -2.146   0.0458 *  
## AgeGroup8w       -2.24498    0.86846  -2.585   0.0187 *  
## ADGKG             0.01751    0.75699   0.023   0.9818    
## AgeGroup2w:ADGKG -0.42295    0.80094  -0.528   0.6039    
## AgeGroup8w:ADGKG  0.86269    0.86550   0.997   0.3321    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for quasipoisson family taken to be 18.86331)
## 
##     Null deviance: 8117.2  on 23  degrees of freedom
## Residual deviance:  348.5  on 18  degrees of freedom
## AIC: NA
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p><br>
Just like the Shannon’s diversity, the interaction terms are not significant, so we’ll remove those and run the glm again.
<br></p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="free-space.html#cb69-1" aria-hidden="true" tabindex="-1"></a>qp.chao.all2 <span class="ot">=</span> <span class="fu">glm</span>(chao <span class="sc">~</span> AgeGroup<span class="sc">+</span>ADGKG, <span class="at">data =</span> meta, <span class="at">family =</span> <span class="st">&quot;quasipoisson&quot;</span>)</span>
<span id="cb69-2"><a href="free-space.html#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(qp.chao.all2)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = chao ~ AgeGroup + ADGKG, family = &quot;quasipoisson&quot;, 
##     data = meta)
## 
## Deviance Residuals: 
##    Min      1Q  Median      3Q     Max  
## -7.783  -3.452  -1.378   3.744   8.184  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  7.03944    0.23567  29.870  &lt; 2e-16 ***
## AgeGroup2w  -1.98090    0.14862 -13.329 2.08e-11 ***
## AgeGroup8w  -1.24286    0.11926 -10.422 1.57e-09 ***
## ADGKG       -0.02643    0.24530  -0.108    0.915    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for quasipoisson family taken to be 23.74583)
## 
##     Null deviance: 8117.20  on 23  degrees of freedom
## Residual deviance:  476.31  on 20  degrees of freedom
## AIC: NA
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p><br>
As we can see, age is now significant for richness aswell.
<br>
We sampled the same animals over time, so this is a repeated measures design. We can add this component with (1|animal) in the lmer function.
<br></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="free-space.html#cb71-1" aria-hidden="true" tabindex="-1"></a>rm.shannon.all <span class="ot">=</span> <span class="fu">lmer</span>(shannon <span class="sc">~</span> AgeGroup<span class="sc">+</span>ADGKG <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Animal), <span class="at">data =</span> meta)</span>
<span id="cb71-2"><a href="free-space.html#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rm.shannon.all)</span></code></pre></div>
<pre><code>## Linear mixed model fit by REML [&#39;lmerMod&#39;]
## Formula: shannon ~ AgeGroup + ADGKG + (1 | Animal)
##    Data: meta
## 
## REML criterion at convergence: 32.4
## 
## Scaled residuals: 
##      Min       1Q   Median       3Q      Max 
## -1.83117 -0.45932  0.09539  0.49972  1.53368 
## 
## Random effects:
##  Groups   Name        Variance Std.Dev.
##  Animal   (Intercept) 0.03793  0.1948  
##  Residual             0.17819  0.4221  
## Number of obs: 24, groups:  Animal, 8
## 
## Fixed effects:
##             Estimate Std. Error t value
## (Intercept)   5.3906     0.3520  15.313
## AgeGroup2w   -3.2739     0.2114 -15.486
## AgeGroup8w   -1.8104     0.2208  -8.201
## ADGKG        -0.1049     0.3321  -0.316
## 
## Correlation of Fixed Effects:
##            (Intr) AgGrp2 AgGrp8
## AgeGroup2w -0.350              
## AgeGroup8w -0.027  0.461       
## ADGKG      -0.884  0.057 -0.293</code></pre>
<p><br>
Very little, 0.0379, of the variance is explained by the animal random effect. This means that we don’t have to include the repeated measures in our final model, but it did need to be checked!
<br>
In conclusion on the alpha diversity: The diversity and richness of the fecal microbiota increases as the dairy cows age. Animal growth, measured by ADG, does not correlate with the diversity or richness of the fecal community.
<br>
<br>
<strong>Looking at the beta diversity</strong> <br>
Beta diversity looks at the dissimilarity <em>between</em> samples, instead of within a sample, like the alpha diversity.
<br>
First, we’ll calculate the non-metric multidimensional scaling using the Bray-Curtis metric. This metric takes into account the presence or absence of OTU’s and the abundance of the OTU’s that are present in the samples. So, this metric looks at richness and diversity.
<br></p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="free-space.html#cb73-1" aria-hidden="true" tabindex="-1"></a>BC.nmds <span class="ot">=</span> <span class="fu">metaMDS</span>(OTU.clean, <span class="at">distance =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">k=</span><span class="dv">2</span>, <span class="at">trymax =</span> <span class="dv">1000</span>)</span></code></pre></div>
<pre><code>## Square root transformation
## Wisconsin double standardization
## Run 0 stress 0.06208119 
## Run 1 stress 0.06210577 
## ... Procrustes: rmse 0.00142196  max resid 0.005448642 
## ... Similar to previous best
## Run 2 stress 0.06208123 
## ... Procrustes: rmse 0.0002939818  max resid 0.0006667466 
## ... Similar to previous best
## Run 3 stress 0.06208111 
## ... New best solution
## ... Procrustes: rmse 0.000204731  max resid 0.000465473 
## ... Similar to previous best
## Run 4 stress 0.0620811 
## ... New best solution
## ... Procrustes: rmse 1.986518e-05  max resid 4.33287e-05 
## ... Similar to previous best
## Run 5 stress 0.06208117 
## ... Procrustes: rmse 5.794984e-05  max resid 0.0001305115 
## ... Similar to previous best
## Run 6 stress 0.06208113 
## ... Procrustes: rmse 3.672359e-05  max resid 8.23805e-05 
## ... Similar to previous best
## Run 7 stress 0.06210572 
## ... Procrustes: rmse 0.001378329  max resid 0.005368037 
## ... Similar to previous best
## Run 8 stress 0.06210378 
## ... Procrustes: rmse 0.001413908  max resid 0.005105586 
## ... Similar to previous best
## Run 9 stress 0.06208125 
## ... Procrustes: rmse 0.0001224896  max resid 0.0002747291 
## ... Similar to previous best
## Run 10 stress 0.06208111 
## ... Procrustes: rmse 1.212188e-05  max resid 2.522293e-05 
## ... Similar to previous best
## Run 11 stress 0.06208121 
## ... Procrustes: rmse 0.0002127404  max resid 0.0004849115 
## ... Similar to previous best
## Run 12 stress 0.06210373 
## ... Procrustes: rmse 0.001425856  max resid 0.005115974 
## ... Similar to previous best
## Run 13 stress 0.3701749 
## Run 14 stress 0.06210382 
## ... Procrustes: rmse 0.001409512  max resid 0.005096637 
## ... Similar to previous best
## Run 15 stress 0.06208124 
## ... Procrustes: rmse 9.356523e-05  max resid 0.0002029152 
## ... Similar to previous best
## Run 16 stress 0.06210382 
## ... Procrustes: rmse 0.001453464  max resid 0.005117477 
## ... Similar to previous best
## Run 17 stress 0.06210562 
## ... Procrustes: rmse 0.001356805  max resid 0.005299213 
## ... Similar to previous best
## Run 18 stress 0.06208121 
## ... Procrustes: rmse 8.948438e-05  max resid 0.0001972643 
## ... Similar to previous best
## Run 19 stress 0.06208123 
## ... Procrustes: rmse 0.0001043305  max resid 0.0002338428 
## ... Similar to previous best
## Run 20 stress 0.06210572 
## ... Procrustes: rmse 0.00138274  max resid 0.005378219 
## ... Similar to previous best
## *** Solution reached</code></pre>
<p><br>
We’ve reached a convergent solution at around 20 iterations, and the stress is very low, 0.06. That means that 2 axis are sufficient to plot our data.
<br>
We’ll plot our data with different colors for each age group.
<br></p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="free-space.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb75-2"><a href="free-space.html#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating a blank plot for the nmds</span></span>
<span id="cb75-3"><a href="free-space.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BC.nmds, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Bray-Curtis&quot;</span>)</span>
<span id="cb75-4"><a href="free-space.html#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding the points</span></span>
<span id="cb75-5"><a href="free-space.html#cb75-5" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(BC.nmds, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup] <span class="co"># [Meta$AgeGroup] buiten haakjes gehaald</span></span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="free-space.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding a legend</span></span>
<span id="cb77-2"><a href="free-space.html#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="fl">5.5</span>, <span class="fl">2.5</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-16-1.png" width="672" />
<br>
We can also calculate the nMDS for the Jaccard metric. This metric only looks at the presence or absence of OTU’s, so it only looks at richness.
<br></p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="free-space.html#cb78-1" aria-hidden="true" tabindex="-1"></a>J.nmds <span class="ot">=</span> <span class="fu">metaMDS</span>(OTU.clean, <span class="at">distance =</span> <span class="st">&quot;jaccard&quot;</span>, <span class="at">k=</span><span class="dv">2</span>, <span class="at">trymax =</span> <span class="dv">1000</span>)</span></code></pre></div>
<pre><code>## Square root transformation
## Wisconsin double standardization
## Run 0 stress 0.06208109 
## Run 1 stress 0.0620812 
## ... Procrustes: rmse 0.000177083  max resid 0.0003999569 
## ... Similar to previous best
## Run 2 stress 0.06210582 
## ... Procrustes: rmse 0.001409216  max resid 0.005429768 
## ... Similar to previous best
## Run 3 stress 0.06210567 
## ... Procrustes: rmse 0.00137946  max resid 0.005369397 
## ... Similar to previous best
## Run 4 stress 0.06208119 
## ... Procrustes: rmse 0.0001744131  max resid 0.0003942546 
## ... Similar to previous best
## Run 5 stress 0.06208115 
## ... Procrustes: rmse 0.000130016  max resid 0.0002909929 
## ... Similar to previous best
## Run 6 stress 0.06208112 
## ... Procrustes: rmse 0.0001066234  max resid 0.0002384603 
## ... Similar to previous best
## Run 7 stress 0.0620811 
## ... Procrustes: rmse 9.468274e-05  max resid 0.0002141725 
## ... Similar to previous best
## Run 8 stress 0.0621059 
## ... Procrustes: rmse 0.001426217  max resid 0.005460826 
## ... Similar to previous best
## Run 9 stress 0.06208124 
## ... Procrustes: rmse 0.0002015441  max resid 0.0004545264 
## ... Similar to previous best
## Run 10 stress 0.0620812 
## ... Procrustes: rmse 0.0001797232  max resid 0.0004060386 
## ... Similar to previous best
## Run 11 stress 0.06208109 
## ... New best solution
## ... Procrustes: rmse 6.139566e-05  max resid 0.0001371443 
## ... Similar to previous best
## Run 12 stress 0.0621038 
## ... Procrustes: rmse 0.001406932  max resid 0.005095216 
## ... Similar to previous best
## Run 13 stress 0.06208126 
## ... Procrustes: rmse 0.0001414878  max resid 0.0003183287 
## ... Similar to previous best
## Run 14 stress 0.06208108 
## ... New best solution
## ... Procrustes: rmse 3.770942e-05  max resid 7.95009e-05 
## ... Similar to previous best
## Run 15 stress 0.06210568 
## ... Procrustes: rmse 0.001379045  max resid 0.005369661 
## ... Similar to previous best
## Run 16 stress 0.06208111 
## ... Procrustes: rmse 8.244032e-05  max resid 0.0001835168 
## ... Similar to previous best
## Run 17 stress 0.06208123 
## ... Procrustes: rmse 0.0001586643  max resid 0.0003545319 
## ... Similar to previous best
## Run 18 stress 0.06208122 
## ... Procrustes: rmse 0.0001644494  max resid 0.0003672165 
## ... Similar to previous best
## Run 19 stress 0.06208112 
## ... Procrustes: rmse 7.816312e-05  max resid 0.0001743423 
## ... Similar to previous best
## Run 20 stress 0.0620811 
## ... Procrustes: rmse 7.313823e-05  max resid 0.0001625506 
## ... Similar to previous best
## *** Solution reached</code></pre>
<p><br>
With this metric, we’ve also reached a convergent solution after 20 iterations. The stress is also very low for this metric, 0.06. So, we’ll plot this metric with 2 axis aswell.
<br></p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="free-space.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(J.nmds, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Jaccard&quot;</span>)</span>
<span id="cb80-2"><a href="free-space.html#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(J.nmds, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="free-space.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="fl">1.5</span>, <span class="at">legend=</span><span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-1.png" width="672" />
<br>
The values in the plot are somewhat different, but the distributions are very similar. That’s because Jaccard = 2xBray-Curtis/(1+Bray-Curtis).
<br>
We can also plot the standard error ellipses of the nMDS data. We’ll plot the 99% confidence interval of the Bray-Curtis metric.
<br></p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="free-space.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BC.nmds, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Bray-Curtis&quot;</span>)</span>
<span id="cb83-2"><a href="free-space.html#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="fl">5.5</span>, <span class="fl">2.5</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb83-3"><a href="free-space.html#cb83-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellipse for 2 weeks</span></span>
<span id="cb83-4"><a href="free-space.html#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ordiellipse</span>(BC.nmds, <span class="at">groups =</span> meta<span class="sc">$</span>AgeGroup, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">kind =</span> <span class="st">&quot;se&quot;</span>, <span class="at">conf =</span> <span class="fl">0.99</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">col =</span> <span class="st">&quot;green&quot;</span>, <span class="at">draw =</span> <span class="st">&quot;polygon&quot;</span>, <span class="at">alpha =</span> <span class="dv">200</span>, <span class="at">show.groups =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>), <span class="at">border =</span> <span class="cn">FALSE</span>)</span>
<span id="cb83-5"><a href="free-space.html#cb83-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellipse for 8 weeks</span></span>
<span id="cb83-6"><a href="free-space.html#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ordiellipse</span>(BC.nmds, <span class="at">groups =</span> meta<span class="sc">$</span>AgeGroup, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">kind =</span> <span class="st">&quot;se&quot;</span>, <span class="at">conf =</span> <span class="fl">0.99</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">draw =</span> <span class="st">&quot;polygon&quot;</span>, <span class="at">alpha =</span> <span class="dv">200</span>, <span class="at">show.groups =</span> <span class="fu">c</span>(<span class="st">&quot;8w&quot;</span>), <span class="at">border =</span> <span class="cn">FALSE</span>)</span>
<span id="cb83-7"><a href="free-space.html#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="co"># ELLipse for 1 year</span></span>
<span id="cb83-8"><a href="free-space.html#cb83-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ordiellipse</span>(BC.nmds, <span class="at">groups =</span> meta<span class="sc">$</span>AgeGroup, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">kind =</span> <span class="st">&quot;se&quot;</span>, <span class="at">conf =</span> <span class="fl">0.99</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">draw =</span> <span class="st">&quot;polygon&quot;</span>, <span class="at">alpha =</span> <span class="dv">200</span>, <span class="at">show.groups =</span> <span class="fu">c</span>(<span class="st">&quot;1yr&quot;</span>), <span class="at">border =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-19-1.png" width="672" />
<br>
If the stress is high, over 0.3, it’s better to increase to 3 axes.
<br>
We’ll first calculate the Bray-Curtis nMDS for a 3-axis plot.
<br></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="free-space.html#cb84-1" aria-hidden="true" tabindex="-1"></a>BC.nmds<span class="fl">.3</span>D <span class="ot">=</span> <span class="fu">metaMDS</span>(OTU.clean, <span class="at">distance =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">k=</span><span class="dv">3</span>, <span class="at">trymax=</span><span class="dv">1000</span>)</span></code></pre></div>
<pre><code>## Square root transformation
## Wisconsin double standardization
## Run 0 stress 0.04686211 
## Run 1 stress 0.04741561 
## Run 2 stress 0.04673316 
## ... New best solution
## ... Procrustes: rmse 0.01078619  max resid 0.03453509 
## Run 3 stress 0.05061342 
## Run 4 stress 0.04740116 
## Run 5 stress 0.04984351 
## Run 6 stress 0.04747471 
## Run 7 stress 0.05226505 
## Run 8 stress 0.0529536 
## Run 9 stress 0.04741183 
## Run 10 stress 0.04575764 
## ... New best solution
## ... Procrustes: rmse 0.03871705  max resid 0.1297458 
## Run 11 stress 0.05084344 
## Run 12 stress 0.04719045 
## Run 13 stress 0.04864532 
## Run 14 stress 0.04750072 
## Run 15 stress 0.04793837 
## Run 16 stress 0.04579474 
## ... Procrustes: rmse 0.004492898  max resid 0.01438546 
## Run 17 stress 0.05069325 
## Run 18 stress 0.04857979 
## Run 19 stress 0.05058111 
## Run 20 stress 0.04859455 
## Run 21 stress 0.04996549 
## Run 22 stress 0.04739851 
## Run 23 stress 0.04747497 
## Run 24 stress 0.04675354 
## Run 25 stress 0.04747484 
## Run 26 stress 0.04861683 
## Run 27 stress 0.04575768 
## ... Procrustes: rmse 0.0001087933  max resid 0.0001902161 
## ... Similar to previous best
## *** Solution reached</code></pre>
<p><br>
We’ll extract the x, y and z for the nMDS, and then plot them.
<br></p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="free-space.html#cb86-1" aria-hidden="true" tabindex="-1"></a>BCxyz <span class="ot">=</span> <span class="fu">scores</span>(BC.nmds<span class="fl">.3</span>D, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb86-2"><a href="free-space.html#cb86-2" aria-hidden="true" tabindex="-1"></a>BCxyz</span></code></pre></div>
<pre><code>##                 NMDS1       NMDS2        NMDS3
## 5017.1yr.F -4.7877193  0.32878559 -0.204579523
## 5017.2w.F   3.1847627  0.05697846  1.484325743
## 5017.8w.F   1.0503489 -2.11932201 -1.217100783
## 5020.1yr.F -4.7466509  0.24479143 -0.001927473
## 5020.2w.F   3.4956397 -1.01621364  1.008439783
## 5020.8w.F   1.5785763 -1.92688899  0.465285369
## 5026.1yr.F -4.7617021  0.20961333  0.209854984
## 5026.2w.F   3.3964195  1.09611959 -0.615338225
## 5026.8w.F   3.1475032  2.07012741  1.477070476
## 5031.1yr.F -4.7920071  0.44153808  0.196685523
## 5031.2w.F   3.3507672  0.48092133 -1.491382237
## 5031.8w.F   0.8516766 -1.63544389  0.254264412
## 5037.1yr.F -4.8418075  0.48746059 -0.004313703
## 5037.2w.F   3.6570393  0.26299035 -0.511188778
## 5037.8w.F   3.1287853 -0.82431199 -0.025514353
## 5041.1yr.F -4.7617047  0.28520072  0.060756551
## 5041.2w.F   3.1646155  2.42901236 -1.216761595
## 5041.8w.F   1.0838533 -2.57296922 -0.236134615
## 5045.1yr.F -4.7419263  0.16694129  0.005568155
## 5045.2w.F   1.5045514  3.11037881 -0.462668248
## 5045.8w.F   1.4782189 -2.16474462 -0.450092185
## 5053.1yr.F -4.8160856  0.39888230 -0.016335985
## 5053.2w.F   3.2925437  2.29778435  0.813017161
## 5053.8w.F   0.8843018 -2.10763164  0.478069543</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="free-space.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ly</span>(<span class="at">x=</span>BCxyz[,<span class="dv">1</span>], <span class="at">y=</span>BCxyz[,<span class="dv">2</span>], <span class="at">z=</span>BCxyz[,<span class="dv">3</span>], <span class="at">type =</span> <span class="st">&quot;scatter3d&quot;</span>, <span class="at">mode =</span> <span class="st">&quot;markers&quot;</span>, <span class="at">color =</span> meta<span class="sc">$</span>AgeGroup, <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div id="htmlwidget-f980a9e3b33c49ea7074" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-f980a9e3b33c49ea7074">{"x":{"visdat":{"47741f5f549a":["function () ","plotlyVisDat"]},"cur_data":"47741f5f549a","attrs":{"47741f5f549a":{"x":[-4.78771931045918,3.18476268753522,1.0503489366724,-4.74665091020587,3.49563969661543,1.5785762791915,-4.76170206868453,3.3964194861303,3.14750317496417,-4.79200712801778,3.35076724688771,0.851676644654691,-4.84180751419167,3.65703934402551,3.12878532308213,-4.7617047247967,3.16461553839772,1.08385331368147,-4.74192629193479,1.50455144673167,1.47821890162157,-4.81608558041068,3.29254366873902,0.884301839770668],"y":[0.328785594747687,0.0569784618449109,-2.11932201408607,0.244791431449645,-1.01621363550618,-1.92688898639618,0.209613326155857,1.09611958519415,2.07012740888467,0.441538083640638,0.480921333582907,-1.63544388612034,0.487460593750542,0.262990349033227,-0.824311993082458,0.285200716116999,2.42901235588814,-2.57296922090692,0.166941287510249,3.11037881450245,-2.16474462015601,0.398882303388391,2.29778435322892,-2.10763164266522],"z":[-0.20457952327386,1.48432574336691,-1.21710078253216,-0.00192747251577329,1.00843978284518,0.465285369061843,0.209854984407134,-0.615338225488531,1.47707047625416,0.196685522708958,-1.49138223669158,0.254264412164606,-0.00431370321553942,-0.511188777772036,-0.025514352591185,0.0607565513943446,-1.21676159479184,-0.236134614700232,0.00556815486584552,-0.462668247727121,-0.450092185135792,-0.0163359845319067,0.813017160751732,0.478069543146837],"mode":"markers","color":["1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w"],"colors":["blue","green","red"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d"}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"xaxis":{"title":[]},"yaxis":{"title":[]},"zaxis":{"title":[]}},"hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[-4.78771931045918,-4.74665091020587,-4.76170206868453,-4.79200712801778,-4.84180751419167,-4.7617047247967,-4.74192629193479,-4.81608558041068],"y":[0.328785594747687,0.244791431449645,0.209613326155857,0.441538083640638,0.487460593750542,0.285200716116999,0.166941287510249,0.398882303388391],"z":[-0.20457952327386,-0.00192747251577329,0.209854984407134,0.196685522708958,-0.00431370321553942,0.0607565513943446,0.00556815486584552,-0.0163359845319067],"mode":"markers","type":"scatter3d","name":"1yr","marker":{"color":"rgba(0,0,255,1)","line":{"color":"rgba(0,0,255,1)"}},"textfont":{"color":"rgba(0,0,255,1)"},"error_y":{"color":"rgba(0,0,255,1)"},"error_x":{"color":"rgba(0,0,255,1)"},"line":{"color":"rgba(0,0,255,1)"},"frame":null},{"x":[3.18476268753522,3.49563969661543,3.3964194861303,3.35076724688771,3.65703934402551,3.16461553839772,1.50455144673167,3.29254366873902],"y":[0.0569784618449109,-1.01621363550618,1.09611958519415,0.480921333582907,0.262990349033227,2.42901235588814,3.11037881450245,2.29778435322892],"z":[1.48432574336691,1.00843978284518,-0.615338225488531,-1.49138223669158,-0.511188777772036,-1.21676159479184,-0.462668247727121,0.813017160751732],"mode":"markers","type":"scatter3d","name":"2w","marker":{"color":"rgba(0,255,0,1)","line":{"color":"rgba(0,255,0,1)"}},"textfont":{"color":"rgba(0,255,0,1)"},"error_y":{"color":"rgba(0,255,0,1)"},"error_x":{"color":"rgba(0,255,0,1)"},"line":{"color":"rgba(0,255,0,1)"},"frame":null},{"x":[1.0503489366724,1.5785762791915,3.14750317496417,0.851676644654691,3.12878532308213,1.08385331368147,1.47821890162157,0.884301839770668],"y":[-2.11932201408607,-1.92688898639618,2.07012740888467,-1.63544388612034,-0.824311993082458,-2.57296922090692,-2.16474462015601,-2.10763164266522],"z":[-1.21710078253216,0.465285369061843,1.47707047625416,0.254264412164606,-0.025514352591185,-0.236134614700232,-0.450092185135792,0.478069543146837],"mode":"markers","type":"scatter3d","name":"8w","marker":{"color":"rgba(255,0,0,1)","line":{"color":"rgba(255,0,0,1)"}},"textfont":{"color":"rgba(255,0,0,1)"},"error_y":{"color":"rgba(255,0,0,1)"},"error_x":{"color":"rgba(255,0,0,1)"},"line":{"color":"rgba(255,0,0,1)"},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p><br>
3D plots are difficult to interpret in articles, so many authors choose to create two 2D plots instead.
<br></p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="free-space.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb89-2"><a href="free-space.html#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting axis 1 and 2, which are x and y</span></span>
<span id="cb89-3"><a href="free-space.html#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BCxyz[,<span class="dv">1</span>], BCxyz[,<span class="dv">2</span>], <span class="at">main =</span> <span class="st">&quot;Bray-Curtis 1:2&quot;</span>, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="free-space.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="fl">5.4</span>, <span class="dv">3</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb91-2"><a href="free-space.html#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting axis 1 and 3, which are x and z</span></span>
<span id="cb91-3"><a href="free-space.html#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BCxyz[,<span class="dv">1</span>], BCxyz[,<span class="dv">3</span>], <span class="at">main =</span> <span class="st">&quot;Bray-Curtis 1:3&quot;</span>, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<p><br>
The most common type of beta-diversity metric is UniFrac. This metric takes phylogenetic relationships into account, as opposed to Bray-Curtis and Jaccard. So, with UniFrac, samples with different OTU’s from the same genus will be more similar than samples with OTU’s from different genera. The drawback of UniFrac is that it’s sensitive to low abundance OTU’s.
<br>
To plot UniFrac, we first have to make a phyloseq object. That means we need OTU.clean, meta and tax.clean data. After we’ve created the three separate objects, we’ll merge them into one.
<br></p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="free-space.html#cb93-1" aria-hidden="true" tabindex="-1"></a>OTU.UF <span class="ot">=</span> <span class="fu">otu_table</span>(<span class="fu">as.matrix</span>(OTU.clean), <span class="at">taxa_are_rows =</span> <span class="cn">FALSE</span>)</span>
<span id="cb93-2"><a href="free-space.html#cb93-2" aria-hidden="true" tabindex="-1"></a>tax.UF <span class="ot">=</span> <span class="fu">tax_table</span>(<span class="fu">as.matrix</span>(tax.clean))</span>
<span id="cb93-3"><a href="free-space.html#cb93-3" aria-hidden="true" tabindex="-1"></a>meta.UF <span class="ot">=</span> <span class="fu">sample_data</span>(meta)</span>
<span id="cb93-4"><a href="free-space.html#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Merging them into an object of class phyloseq</span></span>
<span id="cb93-5"><a href="free-space.html#cb93-5" aria-hidden="true" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">phyloseq</span>(OTU.UF, tax.UF, meta.UF)</span></code></pre></div>
<p><br>
To add the phylogenetic component to UniFrac, we need a rooted phylogenetic tree of the OTU’s. In this workshop, it had already been calculated and stored in the dataset ‘NJ.tree.Rdata’. So, we’ll load this data and add it to the physeq object. Then, we’ll look at the tree and its components.
<br></p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="free-space.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Data/NJ.tree.Rdata&quot;</span>) <span class="co"># Downloaded it from GitHub and stored it in the folder &quot;Data&quot;</span></span>
<span id="cb94-2"><a href="free-space.html#cb94-2" aria-hidden="true" tabindex="-1"></a>physeq.tree <span class="ot">=</span> <span class="fu">merge_phyloseq</span>(physeq, NJ.tree)</span>
<span id="cb94-3"><a href="free-space.html#cb94-3" aria-hidden="true" tabindex="-1"></a>physeq.tree</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 5002 taxa and 24 samples ]
## sample_data() Sample Data:       [ 24 samples by 9 sample variables ]
## tax_table()   Taxonomy Table:    [ 5002 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 5002 tips and 5000 internal nodes ]</code></pre>
<p><br>
Next, we’ll calculate the <em>weighted</em> UniFrac and ordinate it into nMDS.
<br></p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="free-space.html#cb96-1" aria-hidden="true" tabindex="-1"></a>wUF.ordu <span class="ot">=</span> <span class="fu">ordinate</span>(physeq.tree, <span class="at">method =</span> <span class="st">&quot;NMDS&quot;</span>, <span class="at">distance =</span> <span class="st">&quot;unifrac&quot;</span>, <span class="at">weighted =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Run 0 stress 0.08645307 
## Run 1 stress 0.08645305 
## ... New best solution
## ... Procrustes: rmse 1.95726e-05  max resid 5.360447e-05 
## ... Similar to previous best
## Run 2 stress 0.1335688 
## Run 3 stress 0.1462923 
## Run 4 stress 0.08645302 
## ... New best solution
## ... Procrustes: rmse 0.0003084251  max resid 0.0008232998 
## ... Similar to previous best
## Run 5 stress 0.08645296 
## ... New best solution
## ... Procrustes: rmse 0.0002326256  max resid 0.0006198192 
## ... Similar to previous best
## Run 6 stress 0.1157451 
## Run 7 stress 0.1143564 
## Run 8 stress 0.1317675 
## Run 9 stress 0.08645297 
## ... Procrustes: rmse 0.0001396108  max resid 0.0003650213 
## ... Similar to previous best
## Run 10 stress 0.08808604 
## Run 11 stress 0.08645293 
## ... New best solution
## ... Procrustes: rmse 5.607187e-05  max resid 0.000148279 
## ... Similar to previous best
## Run 12 stress 0.115745 
## Run 13 stress 0.08645302 
## ... Procrustes: rmse 0.0001812108  max resid 0.00048756 
## ... Similar to previous best
## Run 14 stress 0.1143564 
## Run 15 stress 0.08659154 
## ... Procrustes: rmse 0.004236648  max resid 0.01797805 
## Run 16 stress 0.1295293 
## Run 17 stress 0.08645307 
## ... Procrustes: rmse 0.0001978456  max resid 0.0005319125 
## ... Similar to previous best
## Run 18 stress 0.1347949 
## Run 19 stress 0.08645294 
## ... Procrustes: rmse 1.302835e-05  max resid 3.061646e-05 
## ... Similar to previous best
## Run 20 stress 0.08808611 
## *** Solution reached</code></pre>
<p><br>
Let’s plot the UniFrac nMDS.
<br></p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="free-space.html#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb98-2"><a href="free-space.html#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wUF.ordu, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Weighted UniFrac&quot;</span>)</span>
<span id="cb98-3"><a href="free-space.html#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(wUF.ordu, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="free-space.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.3</span>, <span class="fl">0.15</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-1.png" width="672" />
<br>
We can also plot it with ggplot2.
<br></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="free-space.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(physeq.tree, wUF.ordu, <span class="at">type =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">color =</span> <span class="st">&quot;AgeGroup&quot;</span>)<span class="sc">+</span></span>
<span id="cb101-2"><a href="free-space.html#cb101-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span><span class="ot">=</span><span class="st">&quot;green&quot;</span>, <span class="st">&quot;8w&quot;</span><span class="ot">=</span><span class="st">&quot;red&quot;</span>, <span class="st">&quot;1yr&quot;</span><span class="ot">=</span><span class="st">&quot;blue&quot;</span>))<span class="sc">+</span></span>
<span id="cb101-3"><a href="free-space.html#cb101-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb101-4"><a href="free-space.html#cb101-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Weighted UniFrac&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-25-1.png" width="672" />
<br>
The Unweighted UniFrac can also be calculated and plotted.
<br></p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="free-space.html#cb102-1" aria-hidden="true" tabindex="-1"></a>uwUF.ordu <span class="ot">=</span> <span class="fu">ordinate</span>(physeq.tree, <span class="at">method =</span> <span class="st">&quot;NMDS&quot;</span>, <span class="at">distance =</span> <span class="st">&quot;unifrac&quot;</span>, <span class="at">weighted =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Run 0 stress 9.695153e-05 
## Run 1 stress 9.657832e-05 
## ... New best solution
## ... Procrustes: rmse 7.750783e-05  max resid 0.0002776914 
## ... Similar to previous best
## Run 2 stress 9.871795e-05 
## ... Procrustes: rmse 8.086551e-05  max resid 0.0002819207 
## ... Similar to previous best
## Run 3 stress 9.488623e-05 
## ... New best solution
## ... Procrustes: rmse 7.261501e-05  max resid 0.0002642816 
## ... Similar to previous best
## Run 4 stress 9.862006e-05 
## ... Procrustes: rmse 1.701217e-05  max resid 5.025527e-05 
## ... Similar to previous best
## Run 5 stress 9.806631e-05 
## ... Procrustes: rmse 0.0001070473  max resid 0.0002353732 
## ... Similar to previous best
## Run 6 stress 9.757454e-05 
## ... Procrustes: rmse 3.985665e-05  max resid 0.0001388531 
## ... Similar to previous best
## Run 7 stress 9.826177e-05 
## ... Procrustes: rmse 9.722135e-05  max resid 0.0002191936 
## ... Similar to previous best
## Run 8 stress 9.695708e-05 
## ... Procrustes: rmse 7.448687e-05  max resid 0.0002751687 
## ... Similar to previous best
## Run 9 stress 9.907648e-05 
## ... Procrustes: rmse 9.310993e-05  max resid 0.0002388289 
## ... Similar to previous best
## Run 10 stress 9.984534e-05 
## ... Procrustes: rmse 3.384419e-05  max resid 0.0001260377 
## ... Similar to previous best
## Run 11 stress 9.684607e-05 
## ... Procrustes: rmse 0.0001319037  max resid 0.0003356478 
## ... Similar to previous best
## Run 12 stress 9.69891e-05 
## ... Procrustes: rmse 8.404145e-06  max resid 2.447679e-05 
## ... Similar to previous best
## Run 13 stress 0.0002969569 
## ... Procrustes: rmse 0.0003866364  max resid 0.0006715474 
## ... Similar to previous best
## Run 14 stress 9.723199e-05 
## ... Procrustes: rmse 3.731826e-05  max resid 0.0001336343 
## ... Similar to previous best
## Run 15 stress 9.99257e-05 
## ... Procrustes: rmse 0.0001270356  max resid 0.0003614341 
## ... Similar to previous best
## Run 16 stress 9.955355e-05 
## ... Procrustes: rmse 6.056256e-05  max resid 0.0001673759 
## ... Similar to previous best
## Run 17 stress 9.589429e-05 
## ... Procrustes: rmse 1.686683e-05  max resid 4.596185e-05 
## ... Similar to previous best
## Run 18 stress 9.633493e-05 
## ... Procrustes: rmse 3.660483e-05  max resid 0.0001324208 
## ... Similar to previous best
## Run 19 stress 9.921893e-05 
## ... Procrustes: rmse 1.085938e-05  max resid 1.669484e-05 
## ... Similar to previous best
## Run 20 stress 9.637055e-05 
## ... Procrustes: rmse 6.450683e-05  max resid 0.0001970587 
## ... Similar to previous best
## *** Solution reached</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="free-space.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(physeq.tree, uwUF.ordu, <span class="at">type =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">color =</span> <span class="st">&quot;AgeGroup&quot;</span>)<span class="sc">+</span></span>
<span id="cb104-2"><a href="free-space.html#cb104-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span> <span class="ot">=</span> <span class="st">&quot;green&quot;</span>, <span class="st">&quot;8w&quot;</span> <span class="ot">=</span> <span class="st">&quot;red&quot;</span>, <span class="st">&quot;1yr&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>))<span class="sc">+</span></span>
<span id="cb104-3"><a href="free-space.html#cb104-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb104-4"><a href="free-space.html#cb104-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Unweighted UniFrac&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-26-1.png" width="672" />
<br>
We can also plot the ellipses of the Weighted UniFrac distances.
<br></p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="free-space.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wUF.ordu, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Weighted UniFrac&quot;</span>)</span>
<span id="cb105-2"><a href="free-space.html#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.3</span>, <span class="fl">0.15</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>,<span class="st">&quot;8w&quot;</span>,<span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;pink&quot;</span>, <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb105-3"><a href="free-space.html#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellipse for 2 weeks</span></span>
<span id="cb105-4"><a href="free-space.html#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ordiellipse</span>(wUF.ordu, <span class="at">groups =</span> meta<span class="sc">$</span>AgeGroup, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">kind =</span> <span class="st">&quot;se&quot;</span>, <span class="at">conf =</span> <span class="fl">0.99</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">col =</span> <span class="st">&quot;pink&quot;</span>, <span class="at">draw =</span> <span class="st">&quot;polygon&quot;</span>, <span class="at">alpha =</span> <span class="dv">200</span>, <span class="at">show.groups =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>), <span class="at">border =</span> <span class="cn">FALSE</span>)</span>
<span id="cb105-5"><a href="free-space.html#cb105-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellipse for 8 weeks</span></span>
<span id="cb105-6"><a href="free-space.html#cb105-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ordiellipse</span>(wUF.ordu, <span class="at">groups =</span> meta<span class="sc">$</span>AgeGroup, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">kind =</span> <span class="st">&quot;se&quot;</span>, <span class="at">conf =</span> <span class="fl">0.99</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">col =</span> <span class="st">&quot;purple&quot;</span>, <span class="at">draw =</span> <span class="st">&quot;polygon&quot;</span>, <span class="at">alpha =</span> <span class="dv">200</span>, <span class="at">show.groups =</span> <span class="fu">c</span>(<span class="st">&quot;8w&quot;</span>), <span class="at">border =</span> <span class="cn">FALSE</span>)</span>
<span id="cb105-7"><a href="free-space.html#cb105-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellipse for 1 year</span></span>
<span id="cb105-8"><a href="free-space.html#cb105-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ordiellipse</span>(wUF.ordu, <span class="at">groups =</span> meta<span class="sc">$</span>AgeGroup, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">kind =</span> <span class="st">&quot;se&quot;</span>, <span class="at">conf =</span> <span class="fl">0.99</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">draw =</span> <span class="st">&quot;polygon&quot;</span>, <span class="at">alpha =</span> <span class="dv">200</span>, <span class="at">show.groups =</span> <span class="fu">c</span>(<span class="st">&quot;1yr&quot;</span>), <span class="at">border =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-27-1.png" width="672" />
<br>
Ellipses can also be plotted with ggplot2, although they look a bit different.
<br></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="free-space.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(physeq.tree, wUF.ordu, <span class="at">type =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">color =</span> <span class="st">&quot;AgeGroup&quot;</span>)<span class="sc">+</span></span>
<span id="cb106-2"><a href="free-space.html#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span> <span class="ot">=</span> <span class="st">&quot;pink&quot;</span>, <span class="st">&quot;8w&quot;</span> <span class="ot">=</span> <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;1yr&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>))<span class="sc">+</span></span>
<span id="cb106-3"><a href="free-space.html#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb106-4"><a href="free-space.html#cb106-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>()<span class="sc">+</span></span>
<span id="cb106-5"><a href="free-space.html#cb106-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Weighted UniFrac&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-28-1.png" width="672" />
<br>
We can calculate UniFrac distances with <em>UniFrac</em> and ordinate it for 3-axes with <em>metaMDS</em>.
<br></p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="free-space.html#cb107-1" aria-hidden="true" tabindex="-1"></a>wUF.dist <span class="ot">=</span> <span class="fu">UniFrac</span>(physeq.tree, <span class="at">weighted =</span> <span class="cn">TRUE</span>, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb107-2"><a href="free-space.html#cb107-2" aria-hidden="true" tabindex="-1"></a>wUF.nmds<span class="fl">.3</span>D <span class="ot">=</span> <span class="fu">metaMDS</span>(wUF.dist, <span class="at">method =</span> <span class="st">&quot;NMDS&quot;</span>, <span class="at">k=</span><span class="dv">3</span>)</span></code></pre></div>
<pre><code>## Run 0 stress 0.04217394 
## Run 1 stress 0.05952403 
## Run 2 stress 0.05952413 
## Run 3 stress 0.04217396 
## ... Procrustes: rmse 4.597236e-05  max resid 0.0001189685 
## ... Similar to previous best
## Run 4 stress 0.04217403 
## ... Procrustes: rmse 0.0001487937  max resid 0.0004561402 
## ... Similar to previous best
## Run 5 stress 0.04217395 
## ... Procrustes: rmse 6.845306e-05  max resid 0.0002343482 
## ... Similar to previous best
## Run 6 stress 0.05952413 
## Run 7 stress 0.042174 
## ... Procrustes: rmse 0.0001351877  max resid 0.0004443447 
## ... Similar to previous best
## Run 8 stress 0.06761596 
## Run 9 stress 0.059524 
## Run 10 stress 0.04217407 
## ... Procrustes: rmse 0.0001709586  max resid 0.0005650254 
## ... Similar to previous best
## Run 11 stress 0.04217396 
## ... Procrustes: rmse 9.078681e-05  max resid 0.0003021391 
## ... Similar to previous best
## Run 12 stress 0.04217405 
## ... Procrustes: rmse 0.000168228  max resid 0.0005161761 
## ... Similar to previous best
## Run 13 stress 0.04217398 
## ... Procrustes: rmse 0.0001098377  max resid 0.0003567109 
## ... Similar to previous best
## Run 14 stress 0.042174 
## ... Procrustes: rmse 8.346652e-05  max resid 0.0002407597 
## ... Similar to previous best
## Run 15 stress 0.04217407 
## ... Procrustes: rmse 0.0001837054  max resid 0.0006272833 
## ... Similar to previous best
## Run 16 stress 0.04217406 
## ... Procrustes: rmse 0.0001481673  max resid 0.0003732481 
## ... Similar to previous best
## Run 17 stress 0.04217395 
## ... Procrustes: rmse 2.316957e-05  max resid 7.009862e-05 
## ... Similar to previous best
## Run 18 stress 0.04217396 
## ... Procrustes: rmse 8.182881e-05  max resid 0.0002803632 
## ... Similar to previous best
## Run 19 stress 0.04217397 
## ... Procrustes: rmse 0.0001031614  max resid 0.0003380335 
## ... Similar to previous best
## Run 20 stress 0.04217399 
## ... Procrustes: rmse 0.0001082464  max resid 0.0003554024 
## ... Similar to previous best
## *** Solution reached</code></pre>
<p><br>
Then, we’ll take out the xyz values and plot them with plotly.
<br></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="free-space.html#cb109-1" aria-hidden="true" tabindex="-1"></a>wUFxyz <span class="ot">=</span> <span class="fu">scores</span>(wUF.nmds<span class="fl">.3</span>D, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>)</span>
<span id="cb109-2"><a href="free-space.html#cb109-2" aria-hidden="true" tabindex="-1"></a>wUFxyz</span></code></pre></div>
<pre><code>##                  NMDS1        NMDS2       NMDS3
## 5017.1yr.F -0.19592858  0.107838852  0.07970378
## 5017.2w.F   0.40344380  0.187070009 -0.11910580
## 5017.8w.F  -0.06745447  0.045811253 -0.21943422
## 5020.1yr.F -0.21314445  0.100936809  0.06830615
## 5020.2w.F  -0.02919525 -0.163696093 -0.02924315
## 5020.8w.F   0.03370365  0.054488455 -0.09105410
## 5026.1yr.F -0.22486455  0.066643683  0.05598675
## 5026.2w.F   0.13248515 -0.217106790  0.08751753
## 5026.8w.F   0.39018092  0.135420606  0.24011653
## 5031.1yr.F -0.19998566  0.080482377  0.09446961
## 5031.2w.F   0.19092361 -0.256894303  0.01561278
## 5031.8w.F  -0.13588109 -0.042296559 -0.02585488
## 5037.1yr.F -0.21803783  0.076509564  0.07189351
## 5037.2w.F   0.05189732 -0.120255191 -0.04231306
## 5037.8w.F   0.14232875 -0.115637316 -0.01896293
## 5041.1yr.F -0.20914084  0.081785024  0.07442644
## 5041.2w.F   0.27828678 -0.237773303  0.03642365
## 5041.8w.F  -0.13941945 -0.001654062 -0.18656492
## 5045.1yr.F -0.23334457  0.051075686  0.06283287
## 5045.2w.F   0.49235803  0.294995885 -0.14636092
## 5045.8w.F  -0.16920077 -0.126231476 -0.13821994
## 5053.1yr.F -0.21544102  0.077998233  0.08008381
## 5053.2w.F   0.27521892 -0.030406547  0.17560574
## 5053.8w.F  -0.13978837 -0.049104795 -0.12586523</code></pre>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="free-space.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ly</span>(<span class="at">x=</span>wUFxyz[,<span class="dv">1</span>], <span class="at">y=</span>wUFxyz[,<span class="dv">2</span>], <span class="at">z=</span>wUFxyz[,<span class="dv">3</span>], <span class="at">type =</span> <span class="st">&quot;scatter3d&quot;</span>, <span class="at">mode =</span> <span class="st">&quot;markers&quot;</span>, <span class="at">color=</span>meta<span class="sc">$</span>AgeGroup, <span class="at">colors=</span><span class="fu">c</span>(<span class="st">&quot;pink&quot;</span>, <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<div id="htmlwidget-ca732052ee48fadf2a1b" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-ca732052ee48fadf2a1b">{"x":{"visdat":{"477468203c53":["function () ","plotlyVisDat"]},"cur_data":"477468203c53","attrs":{"477468203c53":{"x":[-0.19592858103205,0.403443797563631,-0.0674544709789275,-0.213144451318145,-0.029195251142192,0.0337036476920514,-0.224864548726631,0.13248515031895,0.390180917453704,-0.199985662213819,0.19092360734092,-0.13588108508355,-0.218037833728084,0.0518973226399105,0.142328745315163,-0.209140843377632,0.278286780267357,-0.13941944770309,-0.233344570264657,0.492358029925642,-0.169200774321509,-0.215441022562339,0.275218917623397,-0.139788373688101],"y":[0.107838852075638,0.187070009456276,0.0458112527156279,0.100936809207075,-0.163696093319003,0.054488454641568,0.0666436833094421,-0.21710679003518,0.135420605612063,0.0804823765932766,-0.256894303187102,-0.0422965588763816,0.0765095643668562,-0.120255190806781,-0.115637315704089,0.0817850238802328,-0.237773303065919,-0.00165406171854154,0.0510756855539885,0.294995884862548,-0.126231476360129,0.0779982325981863,-0.0304065472094594,-0.0491047945901905],"z":[0.079703776382978,-0.119105795157837,-0.219434218619414,0.0683061522202025,-0.0292431472330305,-0.0910541037239572,0.0559867499766846,0.0875175263883615,0.240116530104297,0.0944696143126626,0.0156127809565279,-0.025854879777142,0.0718935050607912,-0.0423130552214619,-0.0189629299429066,0.0744264359156381,0.0364236489121322,-0.186564923991024,0.0628328700475962,-0.14636091981074,-0.138219938722338,0.0800838092542755,0.175605741577039,-0.125865228909335],"mode":"markers","color":["1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w","1yr","2w","8w"],"colors":["pink","purple","blue"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d"}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"xaxis":{"title":[]},"yaxis":{"title":[]},"zaxis":{"title":[]}},"hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[-0.19592858103205,-0.213144451318145,-0.224864548726631,-0.199985662213819,-0.218037833728084,-0.209140843377632,-0.233344570264657,-0.215441022562339],"y":[0.107838852075638,0.100936809207075,0.0666436833094421,0.0804823765932766,0.0765095643668562,0.0817850238802328,0.0510756855539885,0.0779982325981863],"z":[0.079703776382978,0.0683061522202025,0.0559867499766846,0.0944696143126626,0.0718935050607912,0.0744264359156381,0.0628328700475962,0.0800838092542755],"mode":"markers","type":"scatter3d","name":"1yr","marker":{"color":"rgba(255,192,203,1)","line":{"color":"rgba(255,192,203,1)"}},"textfont":{"color":"rgba(255,192,203,1)"},"error_y":{"color":"rgba(255,192,203,1)"},"error_x":{"color":"rgba(255,192,203,1)"},"line":{"color":"rgba(255,192,203,1)"},"frame":null},{"x":[0.403443797563631,-0.029195251142192,0.13248515031895,0.19092360734092,0.0518973226399105,0.278286780267357,0.492358029925642,0.275218917623397],"y":[0.187070009456276,-0.163696093319003,-0.21710679003518,-0.256894303187102,-0.120255190806781,-0.237773303065919,0.294995884862548,-0.0304065472094594],"z":[-0.119105795157837,-0.0292431472330305,0.0875175263883615,0.0156127809565279,-0.0423130552214619,0.0364236489121322,-0.14636091981074,0.175605741577039],"mode":"markers","type":"scatter3d","name":"2w","marker":{"color":"rgba(160,32,240,1)","line":{"color":"rgba(160,32,240,1)"}},"textfont":{"color":"rgba(160,32,240,1)"},"error_y":{"color":"rgba(160,32,240,1)"},"error_x":{"color":"rgba(160,32,240,1)"},"line":{"color":"rgba(160,32,240,1)"},"frame":null},{"x":[-0.0674544709789275,0.0337036476920514,0.390180917453704,-0.13588108508355,0.142328745315163,-0.13941944770309,-0.169200774321509,-0.139788373688101],"y":[0.0458112527156279,0.054488454641568,0.135420605612063,-0.0422965588763816,-0.115637315704089,-0.00165406171854154,-0.126231476360129,-0.0491047945901905],"z":[-0.219434218619414,-0.0910541037239572,0.240116530104297,-0.025854879777142,-0.0189629299429066,-0.186564923991024,-0.138219938722338,-0.125865228909335],"mode":"markers","type":"scatter3d","name":"8w","marker":{"color":"rgba(0,0,255,1)","line":{"color":"rgba(0,0,255,1)"}},"textfont":{"color":"rgba(0,0,255,1)"},"error_y":{"color":"rgba(0,0,255,1)"},"error_x":{"color":"rgba(0,0,255,1)"},"line":{"color":"rgba(0,0,255,1)"},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p><br>
<br>
It is harder to visualize continuous variables in the way we’ve just visualized categorical values. We can, however, fit these as vectors on the nMDS plots.
<br>
To do that, we’ll first fit the variables to our distances. We’ll use Bray-Curtis and weighted UniFrac for this, but you can do this with Jaccard aswell and you could also use unweighted UniFrac.
<br></p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="free-space.html#cb112-1" aria-hidden="true" tabindex="-1"></a>fit.BC <span class="ot">=</span> <span class="fu">envfit</span>(BC.nmds, meta)</span>
<span id="cb112-2"><a href="free-space.html#cb112-2" aria-hidden="true" tabindex="-1"></a>fit.BC</span></code></pre></div>
<pre><code>## 
## ***VECTORS
## 
##             NMDS1    NMDS2     r2 Pr(&gt;r)    
## AgeExact -0.99893 -0.04630 0.9765  0.001 ***
## ADGKG     0.12541  0.99210 0.0771  0.444    
## chao     -0.98550  0.16970 0.9598  0.001 ***
## shannon  -0.69388  0.72009 0.9469  0.001 ***
## simpson   0.42089 -0.90711 0.7353  0.001 ***
## ace      -0.99737  0.07248 0.9078  0.001 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## Permutation: free
## Number of permutations: 999
## 
## ***FACTORS:
## 
## Centroids:
##                   NMDS1   NMDS2
## Animalcow5017   -0.1827  0.5460
## Animalcow5020    0.0065  0.6579
## Animalcow5026    0.4231 -0.8839
## Animalcow5031   -0.2440  0.1186
## Animalcow5037    0.4943 -0.0574
## Animalcow5041    0.0501 -0.0289
## Animalcow5045   -0.1383 -0.3395
## Animalcow5053   -0.4090 -0.0129
## AgeGroup1yr     -4.4483 -0.1795
## AgeGroup2w       2.5035 -1.0530
## AgeGroup8w       1.9447  1.2325
## AgeGroup.ord2w   2.5035 -1.0530
## AgeGroup.ord8w   1.9447  1.2325
## AgeGroup.ord1yr -4.4483 -0.1795
## 
## Goodness of fit:
##                  r2 Pr(&gt;r)    
## Animal       0.0248  0.997    
## AgeGroup     0.9133  0.001 ***
## AgeGroup.ord 0.9133  0.001 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## Permutation: free
## Number of permutations: 999</code></pre>
<p><br>
R has automatically fit every variable into the meta table.
If we don’t want that, we can tell envfit to only run the variables we want.
<br></p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="free-space.html#cb114-1" aria-hidden="true" tabindex="-1"></a>fit.BC <span class="ot">=</span> <span class="fu">envfit</span>(BC.nmds, meta[,<span class="fu">c</span>(<span class="st">&quot;AgeGroup&quot;</span>, <span class="st">&quot;ADGKG&quot;</span>)])</span>
<span id="cb114-2"><a href="free-space.html#cb114-2" aria-hidden="true" tabindex="-1"></a>fit.BC</span></code></pre></div>
<pre><code>## 
## ***VECTORS
## 
##         NMDS1   NMDS2     r2 Pr(&gt;r)
## ADGKG 0.12541 0.99210 0.0771  0.452
## Permutation: free
## Number of permutations: 999
## 
## ***FACTORS:
## 
## Centroids:
##               NMDS1   NMDS2
## AgeGroup1yr -4.4483 -0.1795
## AgeGroup2w   2.5035 -1.0530
## AgeGroup8w   1.9447  1.2325
## 
## Goodness of fit:
##              r2 Pr(&gt;r)    
## AgeGroup 0.9133  0.001 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## Permutation: free
## Number of permutations: 999</code></pre>
<p><br>
We’ll repeat this for the weighted UniFrac.
<br></p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="free-space.html#cb116-1" aria-hidden="true" tabindex="-1"></a>fit.wUF <span class="ot">=</span> <span class="fu">envfit</span>(wUF.ordu, meta[,<span class="fu">c</span>(<span class="st">&quot;AgeGroup&quot;</span>, <span class="st">&quot;ADGKG&quot;</span>)])</span>
<span id="cb116-2"><a href="free-space.html#cb116-2" aria-hidden="true" tabindex="-1"></a>fit.wUF</span></code></pre></div>
<pre><code>## 
## ***VECTORS
## 
##          NMDS1    NMDS2     r2 Pr(&gt;r)
## ADGKG -0.17839  0.98396 0.0399   0.66
## Permutation: free
## Number of permutations: 999
## 
## ***FACTORS:
## 
## Centroids:
##               NMDS1   NMDS2
## AgeGroup1yr -0.1076 -0.0833
## AgeGroup2w   0.1432  0.0322
## AgeGroup8w  -0.0356  0.0512
## 
## Goodness of fit:
##              r2 Pr(&gt;r)    
## AgeGroup 0.5588  0.001 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## Permutation: free
## Number of permutations: 999</code></pre>
<p><br>
We can plot the 2D nMDS with an arrow for the ADG, eventho the ADG isn’t a significant variable. Thus, you wouldn’t use this figure in a publication or such. We’ll do this with the Bray-Curtis dissimilarity.
<br></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="free-space.html#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BC.nmds, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Bray-Curtis&quot;</span>)</span>
<span id="cb118-2"><a href="free-space.html#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(BC.nmds, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;pink&quot;</span>, <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;blue&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="free-space.html#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="dv">6</span>, <span class="dv">2</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;purple&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;pink&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb120-2"><a href="free-space.html#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit.BC, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>) <span class="co"># Adding the fitted variables</span></span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-34-1.png" width="672" />
<br>
We can also only plot variables with p &lt; 0.05. That’d mean we’d only see the centroids.
<br></p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="free-space.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BC.nmds, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Bray-Curtis&quot;</span>)</span>
<span id="cb121-2"><a href="free-space.html#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(BC.nmds, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="free-space.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="dv">6</span>, <span class="dv">2</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb123-2"><a href="free-space.html#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit.BC, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">p.max =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-35-1.png" width="672" />
<br>
We can also plot the weighted UniFrac.
<br></p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="free-space.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wUF.ordu, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Weighted UniFrac&quot;</span>)</span>
<span id="cb124-2"><a href="free-space.html#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(wUF.ordu, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="free-space.html#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(.<span class="dv">3</span>, .<span class="dv">15</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb126-2"><a href="free-space.html#cb126-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit.wUF, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-36-1.png" width="672" />
<br>
If we fit the OTU.clean data frame to the nMDS, we can add arrows for specific OTU’s in the plot. <br>
If an OTU arrow goes in the same direction as an age group centroid, that OTU tends to increase in abundance in that age group. <br>
Conversely, if an OTU arrow moves in the opposite direction of an age group centroid, that OTU probably decreases in abundance in that age group. <br>
We’ll only fit the first 10 OTU’s from our table, because it takes a long time. <br>
Then, we’ll only plot the significant arrows.
<br></p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="free-space.html#cb127-1" aria-hidden="true" tabindex="-1"></a>fit.BC.OTU <span class="ot">=</span> <span class="fu">envfit</span>(BC.nmds, OTU.clean[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span>
<span id="cb127-2"><a href="free-space.html#cb127-2" aria-hidden="true" tabindex="-1"></a>fit.BC.OTU</span></code></pre></div>
<pre><code>## 
## ***VECTORS
## 
##             NMDS1    NMDS2     r2 Pr(&gt;r)    
## Otu00001  0.71703 -0.69704 0.2478  0.033 *  
## Otu00002  0.46962 -0.88287 0.2108  0.057 .  
## Otu00003  0.25731 -0.96633 0.2503  0.021 *  
## Otu00004  0.25055  0.96810 0.2740  0.030 *  
## Otu00005  0.15507  0.98790 0.2906  0.003 ** 
## Otu00006 -0.96828  0.24987 0.6742  0.001 ***
## Otu00007  0.18000 -0.98367 0.2487  0.009 ** 
## Otu00008  0.40221  0.91555 0.3108  0.016 *  
## Otu00009  0.26281 -0.96485 0.1893  0.062 .  
## Otu00010  0.33879 -0.94086 0.1552  0.078 .  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## Permutation: free
## Number of permutations: 999</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="free-space.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BC.nmds, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Bray-Curtis&quot;</span>)</span>
<span id="cb129-2"><a href="free-space.html#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(BC.nmds, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="free-space.html#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="dv">6</span>, <span class="sc">-</span><span class="fl">1.1</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb131-2"><a href="free-space.html#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit.BC.OTU, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">p.max =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-37-1.png" width="672" />
<br>
We can also plot summed genera or family groups of OTU’s.
<br></p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="free-space.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting all the OTU&#39;s within the Ruminococcus genus</span></span>
<span id="cb132-2"><a href="free-space.html#cb132-2" aria-hidden="true" tabindex="-1"></a>OTU.rumino <span class="ot">=</span> OTU.clean[,tax.clean<span class="sc">$</span>Genus <span class="sc">==</span> <span class="st">&quot;g__Ruminococcus&quot;</span>]</span>
<span id="cb132-3"><a href="free-space.html#cb132-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Summing the abundances of the OTU&#39;s into one variable/column</span></span>
<span id="cb132-4"><a href="free-space.html#cb132-4" aria-hidden="true" tabindex="-1"></a>OTU.rumino<span class="sc">$</span>Rumino.sum <span class="ot">=</span> <span class="fu">rowSums</span>(OTU.rumino)</span>
<span id="cb132-5"><a href="free-space.html#cb132-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Fitting the Ruminococceae group</span></span>
<span id="cb132-6"><a href="free-space.html#cb132-6" aria-hidden="true" tabindex="-1"></a>fit.BC.rumino <span class="ot">=</span> <span class="fu">envfit</span>(BC.nmds, OTU.rumino<span class="sc">$</span>Rumino.sum)</span>
<span id="cb132-7"><a href="free-space.html#cb132-7" aria-hidden="true" tabindex="-1"></a>fit.BC.rumino</span></code></pre></div>
<pre><code>## 
## ***VECTORS
## 
##         NMDS1    NMDS2     r2 Pr(&gt;r)    
## [1,] -0.14496  0.98944 0.6622  0.001 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## Permutation: free
## Number of permutations: 999</code></pre>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="free-space.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting it</span></span>
<span id="cb134-2"><a href="free-space.html#cb134-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BC.nmds, <span class="at">type =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Bray-Curtis&quot;</span>)</span>
<span id="cb134-3"><a href="free-space.html#cb134-3" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(BC.nmds, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">display =</span> <span class="st">&quot;sites&quot;</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))[meta<span class="sc">$</span>AgeGroup]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="free-space.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="sc">-</span><span class="dv">6</span>, <span class="sc">-</span><span class="fl">1.1</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;2w&quot;</span>, <span class="st">&quot;8w&quot;</span>, <span class="st">&quot;1yr&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>), <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb136-2"><a href="free-space.html#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding the fitted variables</span></span>
<span id="cb136-3"><a href="free-space.html#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit.BC.rumino, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Ruminococcus&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cv.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="using-github-and-zotero.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
